<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>TreeProfiler Tutorial &#8212; treeprofiler [1.2.2-beta] documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=d1102ebc" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=12dfc556" />
    <script src="../_static/documentation_options.js?v=fbc843e6"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Tutorial" href="index.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="treeprofiler-tutorial">
<h1><a class="toc-backref" href="#id2" role="doc-backlink">TreeProfiler Tutorial</a><a class="headerlink" href="#treeprofiler-tutorial" title="Link to this heading">¶</a></h1>
</section>
<section id="introduction">
<h1><a class="toc-backref" href="#id3" role="doc-backlink">Introduction</a><a class="headerlink" href="#introduction" title="Link to this heading">¶</a></h1>
<p>TreeProfiler is command-line tool for profiling metadata table into phylogenetic tree with descriptive analysis and output visualization</p>
</section>
<section id="table-of-contents">
<h1><a class="toc-backref" href="#id4" role="doc-backlink">Table of Contents</a><a class="headerlink" href="#table-of-contents" title="Link to this heading">¶</a></h1>
<nav class="contents" id="contents">
<p class="topic-title">Contents</p>
<ul class="simple">
<li><p><a class="reference internal" href="#treeprofiler-tutorial" id="id2">TreeProfiler Tutorial</a></p></li>
<li><p><a class="reference internal" href="#introduction" id="id3">Introduction</a></p></li>
<li><p><a class="reference internal" href="#table-of-contents" id="id4">Table of Contents</a></p></li>
<li><p><a class="reference internal" href="#installation" id="id5">Installation</a></p>
<ul>
<li><p><a class="reference internal" href="#dependencies" id="id6">Dependencies</a></p></li>
<li><p><a class="reference internal" href="#quick-install-via-pip" id="id7">Quick install via pip</a></p></li>
<li><p><a class="reference internal" href="#quick-start-with-examples-dataset" id="id8">Quick Start with examples dataset</a></p></li>
<li><p><a class="reference internal" href="#manual-installation" id="id9">Manual installation</a></p>
<ul>
<li><p><a class="reference internal" href="#install-ete-v4" id="id10">Install ETE v4</a></p></li>
<li><p><a class="reference internal" href="#install-treeprofiler" id="id11">Install TreeProfiler</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#input-files" id="id12">Input files</a></p></li>
<li><p><a class="reference internal" href="#basic-usage" id="id13">Basic Usage</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#using-treeprofiler" id="id14">Using TreeProfiler</a></p>
<ul>
<li><p><a class="reference internal" href="#reading-input-tree" id="id15">Reading input tree</a></p>
<ul>
<li><p><a class="reference internal" href="#tree-format" id="id16">Tree format</a></p></li>
<li><p><a class="reference internal" href="#tree-parser" id="id17">Tree parser</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#treeprofiler-annotate-computing-phylogenetic-profiles-and-annotation" id="id18"><code class="docutils literal notranslate"><span class="pre">treeprofiler-annotate</span></code> computing phylogenetic profiles and annotation</a></p>
<ul>
<li><p><a class="reference internal" href="#annotate-metadata-into-tree" id="id19">Annotate metadata into tree</a></p>
<ul>
<li><p><a class="reference internal" href="#basic-metadata-in-tsv-csv-format" id="id20">Basic metadata in TSV/CSV format</a></p></li>
<li><p><a class="reference internal" href="#metadata-tsv-csv-as-a-numerical-data-matrix" id="id21">Metadata TSV/CSV as a numerical data matrix</a></p></li>
<li><p><a class="reference internal" href="#metadata-without-column-names" id="id22">Metadata without column names</a></p></li>
<li><p><a class="reference internal" href="#missing-value-detection" id="id23">Missing value detection</a></p></li>
<li><p><a class="reference internal" href="#unmapped-tree-leaf-property-detection" id="id24">Unmapped tree leaf property detection</a></p></li>
<li><p><a class="reference internal" href="#handling-duplicated-leaf-names-in-metadata" id="id25">Handling duplicated leaf names in metadata</a></p></li>
<li><p><a class="reference internal" href="#specifying-data-types-for-metadata-columns" id="id26">Specifying data types for metadata columns</a></p></li>
<li><p><a class="reference internal" href="#annotate-metadata-to-internal-nodes" id="id27">Annotate metadata to internal nodes</a></p>
<ul>
<li><p><a class="reference internal" href="#customizing-summary-methods-for-different-columns" id="id28">Customizing summary methods for different columns</a></p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference internal" href="#taxonomic-annotation" id="id29">Taxonomic annotation</a></p>
<ul>
<li><p><a class="reference internal" href="#using-different-taxonomic-databases-from-gtdb-ncbi" id="id30">Using different taxonomic databases from GTDB/NCBI</a></p></li>
<li><p><a class="reference internal" href="#identifying-taxon-names-in-different-metadata-fields-columns" id="id31">Identifying taxon names in different metadata fields/columns</a></p></li>
<li><p><a class="reference internal" href="#ignoring-unclassified-taxonomic-annotations" id="id32">Ignoring unclassified taxonomic annotations</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#functional-annotation" id="id33">Functional annotation</a></p>
<ul>
<li><p><a class="reference internal" href="#annotating-multiple-sequence-alignments-msa" id="id34">Annotating Multiple Sequence Alignments(MSA)</a></p></li>
<li><p><a class="reference internal" href="#annotating-eggnog-mapper-output-and-domain-information" id="id35">Annotating EggNOG-Mapper output and Domain information</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#analytical-methods-in-treeprofiler" id="id36">Analytical methods in treeProfiler</a></p>
<ul>
<li><p><a class="reference internal" href="#ancestral-character-reconstruction" id="id37">Ancestral character reconstruction</a></p></li>
<li><p><a class="reference internal" href="#phylogenetic-signal-delta-statistic" id="id38">Phylogenetic signal delta statistic</a></p></li>
<li><p><a class="reference internal" href="#lineage-specificity-analysis" id="id39">Lineage specificity analysis</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output-formats-for-annotated-trees" id="id40">Output formats for annotated trees</a></p></li>
</ul>
</li>
</ul>
</li>
</ul>
</nav>
</section>
<section id="installation">
<h1><a class="toc-backref" href="#id5" role="doc-backlink">Installation</a><a class="headerlink" href="#installation" title="Link to this heading">¶</a></h1>
<section id="dependencies">
<h2><a class="toc-backref" href="#id6" role="doc-backlink">Dependencies</a><a class="headerlink" href="#dependencies" title="Link to this heading">¶</a></h2>
<dl class="simple">
<dt>TreeProfiler requires</dt><dd><ul class="simple">
<li><p>Python version &gt;= 3.9</p></li>
<li><p>ETE Toolkit v4</p></li>
<li><p>biopython</p></li>
<li><p>selenium</p></li>
<li><p>scipy</p></li>
<li><p>matplotlib</p></li>
<li><p>numba</p></li>
<li><p>pastml (custom)</p></li>
</ul>
</dd>
</dl>
</section>
<section id="quick-install-via-pip">
<h2><a class="toc-backref" href="#id7" role="doc-backlink">Quick install via pip</a><a class="headerlink" href="#quick-install-via-pip" title="Link to this heading">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Install ETE Toolkit v4</span>
<span class="n">pip</span> <span class="n">install</span> <span class="o">--</span><span class="n">force</span><span class="o">-</span><span class="n">reinstall</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">etetoolkit</span><span class="o">/</span><span class="n">ete</span><span class="o">/</span><span class="n">archive</span><span class="o">/</span><span class="n">ete4</span><span class="o">.</span><span class="n">zip</span>

<span class="c1"># Install TreeProfiler dependencies</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">biopython</span> <span class="n">selenium</span> <span class="n">scipy</span> <span class="n">matplotlib</span> <span class="n">numba</span>

<span class="c1"># Install custom pastml package for ete4</span>
<span class="n">pip</span> <span class="n">install</span> <span class="s2">&quot;git+https://github.com/dengzq1234/pastml.git@pastml2ete4&quot;</span>

<span class="c1"># Install TreeProfiler tool via pip</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">TreeProfiler</span>

<span class="c1"># or installing main repo</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">compgenomicslab</span><span class="o">/</span><span class="n">TreeProfiler</span><span class="o">/</span><span class="n">archive</span><span class="o">/</span><span class="n">main</span><span class="o">.</span><span class="n">zip</span>
<span class="c1"># or development mode for latestest update</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">git</span><span class="o">+</span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">compgenomicslab</span><span class="o">/</span><span class="n">TreeProfiler</span><span class="nd">@dev</span><span class="o">-</span><span class="n">repo</span>
</pre></div>
</div>
</section>
<section id="quick-start-with-examples-dataset">
<h2><a class="toc-backref" href="#id8" role="doc-backlink">Quick Start with examples dataset</a><a class="headerlink" href="#quick-start-with-examples-dataset" title="Link to this heading">¶</a></h2>
<p>TreeProfiler provide various example dataset for testing in <code class="docutils literal notranslate"><span class="pre">examples/</span></code> or <a class="reference external" href="https://github.com/compgenomicslab/TreeProfiler/tree/main/examples">https://github.com/compgenomicslab/TreeProfiler/tree/main/examples</a>, each directory consists a demo script <code class="docutils literal notranslate"><span class="pre">*_demo.sh</span></code> for quick starting different functions in TreeProfiler which already has annotate-plot pipeline of example data. User can fast explore different example tree with different visualizations. Here is the demonstration:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># execute demo script of example1</span>
<span class="n">cd</span> <span class="n">examples</span><span class="o">/</span><span class="n">basic_example1</span><span class="o">/</span>
<span class="n">sh</span> <span class="o">./</span><span class="n">example1_demo</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>This script will annotate example tree with two metadata tables and visualize the tree with different properties. The output will be shown in the terminal and a local server will be launched for interactive visualization.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>  start parsing...
  Time for parse_csv to run:  0.001968860626220703
  Time for load_metadata_to_tree to run:  0.0003094673156738281
  Time for merge annotations to run:  0.05160331726074219
  Time for annotate_taxa to run:  4.76837158203125e-07
  Visualize properties categorical data random_type in rectangle_layout, numerical data sample1, sample2 in heatmap_layout and barplot_layout.
  Current trees in memory: 0
  Added tree example with id 0.
  * Serving Flask app &#39;ete4.smartview.gui.server&#39; (lazy loading)
  * Environment: production
    WARNING: This is a development server. Do not use it in a production deployment.
    Use a production WSGI server instead.
  * Debug mode: on
  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)


As the session starts in local server http://127.0.0.1:5000, annotated tree and selected properties are visualized at the interactive session. ![treeprofiler interface](https://github.com/dengzq1234/treeprofiler_gallery/blob/main/figure1_all.png?raw=true) Here is detailed introduction of interactive session of visualization([here](#interactive-visualization-interface))
</pre></div>
</div>
<img alt="treeprofiler interface" class="align-center" src="https://github.com/dengzq1234/treeprofiler_gallery/blob/main/figure1_all.png?raw=true" />
<p>Here is detailed introduction of interactive session of visualization([here](#interactive-visualization-interface))</p>
<p>Check other tutorial scripts</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># display demo script of each example</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">basic_example1</span><span class="o">/</span><span class="n">example1_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">automatic_query</span><span class="o">/</span><span class="n">highlight_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">automatic_query</span><span class="o">/</span><span class="n">collapse_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">automatic_query</span><span class="o">/</span><span class="n">prune_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">basic_example2</span><span class="o">/</span><span class="n">example2_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">taxonomy_example</span><span class="o">/</span><span class="n">ncbi</span><span class="o">/</span><span class="n">ncbi_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">taxonomy_example</span><span class="o">/</span><span class="n">gtdb</span><span class="o">/</span><span class="n">gtdb_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">pratical_example</span><span class="o">/</span><span class="n">progenome3</span><span class="o">/</span><span class="n">progenome_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">pratical_example</span><span class="o">/</span><span class="n">gtdb_r202</span><span class="o">/</span><span class="n">gtdbv202full_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">pratical_example</span><span class="o">/</span><span class="n">gtdb_r202</span><span class="o">/</span><span class="n">gtdbv202lite_demo</span><span class="o">.</span><span class="n">sh</span>
<span class="o">./</span><span class="n">examples</span><span class="o">/</span><span class="n">pratical_example</span><span class="o">/</span><span class="n">emapper</span><span class="o">/</span><span class="n">emapper_demo</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
</section>
<section id="manual-installation">
<h2><a class="toc-backref" href="#id9" role="doc-backlink">Manual installation</a><a class="headerlink" href="#manual-installation" title="Link to this heading">¶</a></h2>
<section id="install-ete-v4">
<h3><a class="toc-backref" href="#id10" role="doc-backlink">Install ETE v4</a><a class="headerlink" href="#install-ete-v4" title="Link to this heading">¶</a></h3>
<p>Quick way</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">etetoolkit</span><span class="o">/</span><span class="n">ete</span><span class="o">/</span><span class="n">archive</span><span class="o">/</span><span class="n">ete4</span><span class="o">.</span><span class="n">zip</span>
</pre></div>
</div>
<p>For local development
To install ETE in a local directory to help with the development, you can:</p>
<ul>
<li><p>Clone this repository (git clone <a class="reference external" href="https://github.com/etetoolkit/ete.git">https://github.com/etetoolkit/ete.git</a>)</p></li>
<li><p>Install dependecies
- If you are using conda:</p>
<p><code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">install</span> <span class="pre">-c</span> <span class="pre">conda-forge</span> <span class="pre">cython</span> <span class="pre">bottle</span> <span class="pre">brotli</span> <span class="pre">numpy</span> <span class="pre">pyqt</span></code></p>
<ul class="simple">
<li><p>Otherwise, you can install them with</p></li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">&lt;dependencies&gt;</span></code></p>
<ul class="simple">
<li><p>Build and install ete4 from the repository’s root directory:</p></li>
</ul>
<p><code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">-e</span> <span class="pre">.</span></code></p>
</li>
</ul>
<p>(In Linux there may be some cases where the gcc library must be installed, which can be done with <code class="docutils literal notranslate"><span class="pre">conda</span> <span class="pre">install</span> <span class="pre">-c</span> <span class="pre">conda-forge</span> <span class="pre">gcc_linux-64</span></code>)</p>
</section>
<section id="install-treeprofiler">
<h3><a class="toc-backref" href="#id11" role="doc-backlink">Install TreeProfiler</a><a class="headerlink" href="#install-treeprofiler" title="Link to this heading">¶</a></h3>
<p>Install dependencies</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># install BioPython, selenium, scipy via conda</span>
<span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="n">biopython</span> <span class="n">selenium</span> <span class="n">scipy</span> <span class="n">matplotlib</span>

<span class="c1"># or pip</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">biopython</span> <span class="n">selenium</span> <span class="n">scipy</span> <span class="n">matplotlib</span>
</pre></div>
</div>
<p>Install TreeProfiler</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># install TreeProfiler</span>
<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">compgenomicslab</span><span class="o">/</span><span class="n">TreeProfiler</span>
<span class="n">cd</span> <span class="n">TreeProfiler</span><span class="o">/</span>
<span class="n">python</span> <span class="n">setup</span><span class="o">.</span><span class="n">py</span> <span class="n">install</span>
</pre></div>
</div>
<p>or install inrectly from github</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># install directly</span>
<span class="n">pip</span> <span class="n">install</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">dengzq1234</span><span class="o">/</span><span class="n">TreeProfiler</span><span class="o">/</span><span class="n">archive</span><span class="o">/</span><span class="n">refs</span><span class="o">/</span><span class="n">tags</span><span class="o">/</span><span class="n">v1</span><span class="mf">.1.0</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
</section>
</section>
<section id="input-files">
<h2><a class="toc-backref" href="#id12" role="doc-backlink">Input files</a><a class="headerlink" href="#input-files" title="Link to this heading">¶</a></h2>
<p>TreeProfiler takes following file types as input</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Input</p></th>
<th class="head"><p>Filetype</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Tree</p></td>
<td><p>newick, ete</p></td>
</tr>
<tr class="row-odd"><td><p>Metadata</p></td>
<td><p>tar.gz, tsv</p></td>
</tr>
</tbody>
</table>
<ul class="simple">
<li><p>ete format is a novel format developed to solve the situation we encounter in the previous step, annotated tree can be recover easily with all the annotated data without changing the data type. Besides, the ete format optimized the tree file size after mapped with its associated data. Hence it’s very handy for programers in their own script. At this moment we can only view the ete format in treeprofiler, but we will make the ete format more universal to other phylogenetic software.</p></li>
<li><p>Metadata input could be single or multiple files, either tar.gz compressed file(s) which contains multiple .tsv or plain .tsv file(s).</p></li>
</ul>
</section>
<section id="basic-usage">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">Basic Usage</a><a class="headerlink" href="#basic-usage" title="Link to this heading">¶</a></h2>
<dl class="simple">
<dt>TreeProfiler has two main subcommand:</dt><dd><ul class="simple">
<li><p>annotate</p></li>
<li><p>plot</p></li>
</ul>
</dd>
</dl>
<p>The first one <code class="docutils literal notranslate"><span class="pre">annotate</span></code> is used to annotate your input tree and corresponding metadata, TreeProfiler will map all the metadata into corresponding tree node. In this step, annotated tree will be generated in newick and ete format</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">treeprofiler</span> <span class="n">annotate</span> <span class="o">--</span><span class="n">tree</span> <span class="n">tree</span><span class="o">.</span><span class="n">nw</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">type</span> <span class="n">newick</span> <span class="o">--</span><span class="n">metadata</span> <span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">outdir</span> <span class="o">./</span>
</pre></div>
</div>
<p>The second subcommand <code class="docutils literal notranslate"><span class="pre">plot</span></code> is used to visualize tree with associated metadata. By default, treeprofiler will launch an interactive session at localhost for user to explore input tree.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># plot tree with newick format</span>
<span class="n">treeprofiler</span> <span class="n">plot</span> <span class="o">--</span><span class="n">tree</span> <span class="n">tree_annotated</span><span class="o">.</span><span class="n">nw</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">type</span> <span class="n">newick</span>

<span class="c1"># plot tree with ete format</span>
<span class="n">treeprofiler</span> <span class="n">plot</span> <span class="o">--</span><span class="n">tree</span> <span class="n">tree_annotated</span><span class="o">.</span><span class="n">ete</span> <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">type</span> <span class="n">ete</span>
</pre></div>
</div>
</section>
</section>
<section id="using-treeprofiler">
<h1><a class="toc-backref" href="#id14" role="doc-backlink">Using TreeProfiler</a><a class="headerlink" href="#using-treeprofiler" title="Link to this heading">¶</a></h1>
<p>In this Tutorial we will use TreeProfiler and demostrate basic usage with data in <code class="docutils literal notranslate"><span class="pre">examples/</span></code></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>tree examples/
examples/
├── automatic_query
│   ├── basic_example1_metadata1.tsv
│   ├── basic_example1.nw
│   ├── collapse_demo.sh
│   ├── highlight_demo.sh
│   └── prune_demo.sh
├── basic_example1
│   ├── basic_example1_metadata1.tsv
│   ├── basic_example1_metadata2.tsv
│   ├── basic_example1.nw
│   └── example1_demo.sh
├── basic_example2
│   ├── diauxic.array
│   ├── diauxic.nw
│   ├── example2_demo.sh
│   ├── FluA_H3_AA.fas
│   ├── MCC_FluA_H3_Genotype.txt
│   └── MCC_FluA_H3.nw
├── pratical_example
│   ├── emapper
│   │   ├── 7955.ENSDARP00000116736.aln.faa
│   │   ├── 7955.ENSDARP00000116736.nw
│   │   ├── 7955.out.emapper.annotations
│   │   ├── 7955.out.emapper.pfam
│   │   ├── 7955.out.emapper.smart.out
│   │   ├── emapper_demo.sh
│   │   ├── nifH.faa.aln
│   │   ├── nifH.nw
│   │   ├── nifH.out.emapper.annotations
│   │   └── nifH.out.emapper.pfam
│   ├── gtdb_r202
│   │   ├── ar122_metadata_r202_lite.tar.gz
│   │   ├── bac120_metadata_r202_lite.tar.gz
│   │   ├── gtdbv202full_demo.sh
│   │   ├── gtdbv202lite_demo.sh
│   │   ├── gtdbv202.nw
│   │   ├── merge_gtdbtree.py
│   │   └── progenome3.tar.gz
│   └── progenome3
│       ├── progenome3.nw
│       ├── progenome3.tsv
│       └── progenome_demo.sh
└── taxonomy_example
    ├── gtdb
    │   ├── gtdb_demo.sh
    │   ├── gtdb_example1.nw
    │   └── gtdb_example1.tsv
    └── ncbi
        ├── ncbi_demo.sh
        ├── ncbi_example.nw
        └── ncbi_example.tsv
</pre></div>
</div>
<section id="reading-input-tree">
<h2><a class="toc-backref" href="#id15" role="doc-backlink">Reading input tree</a><a class="headerlink" href="#reading-input-tree" title="Link to this heading">¶</a></h2>
<section id="tree-format">
<h3><a class="toc-backref" href="#id16" role="doc-backlink">Tree format</a><a class="headerlink" href="#tree-format" title="Link to this heading">¶</a></h3>
<p>TreeProfiler accpept input tree in <code class="docutils literal notranslate"><span class="pre">.nw</span></code> or <code class="docutils literal notranslate"><span class="pre">.ete</span></code> by putting <code class="docutils literal notranslate"><span class="pre">--input-type</span> <span class="pre">{newick,ete}</span></code> flag to identify. By default, TreeProfiler will automatically detech the format of tree. The difference between <code class="docutils literal notranslate"><span class="pre">.nw</span></code> and <code class="docutils literal notranslate"><span class="pre">.ete</span></code>:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">newick</span></code> format is more universal and be able to used in different other phylogenetic software although associated data of tree nodes will be considered as plain text.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ete</span></code> format is a novel format developed to solve the situation we encounter in the previous step, annotated tree can be <strong>recover easily with all the annotated data without changing the data type</strong>. Besides, the ete format optimized the tree file size after mapped with its associated data. Hence it’s very handy for programers in their own script. At this moment we can only view the ete format in treeprofiler, but we will make the ete format more universal to other phylogenetic software. <strong>Hence using ete format in ``plot`` subcommand is highly reccomended</strong></p></li>
</ul>
</div></blockquote>
</section>
<section id="tree-parser">
<h3><a class="toc-backref" href="#id17" role="doc-backlink">Tree parser</a><a class="headerlink" href="#tree-parser" title="Link to this heading">¶</a></h3>
<p>TreeProfiler provides argument <code class="docutils literal notranslate"><span class="pre">--internal</span> <span class="pre">{name,support}</span></code> to specify <code class="docutils literal notranslate"><span class="pre">newick</span></code> tree when it include values in internal node. <code class="docutils literal notranslate"><span class="pre">[default:</span> <span class="pre">name]</span></code></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>newick</p></th>
<th class="head"><p>leaves</p></th>
<th class="head"><p>internal_node value</p></th>
<th class="head"><p>internal_parser</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>(A:0.5, B:0.5)Internal_C:0.5;</p></td>
<td><p>A, B</p></td>
<td><p>Internal_C</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">name</span></code></p></td>
</tr>
<tr class="row-odd"><td><p>(A:0.5, B:0.5)0.99:0.5;</p></td>
<td><p>A, B</p></td>
<td><p>0.99</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">support</span></code></p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="treeprofiler-annotate-computing-phylogenetic-profiles-and-annotation">
<h2><a class="toc-backref" href="#id18" role="doc-backlink"><code class="docutils literal notranslate"><span class="pre">treeprofiler-annotate</span></code> computing phylogenetic profiles and annotation</a><a class="headerlink" href="#treeprofiler-annotate-computing-phylogenetic-profiles-and-annotation" title="Link to this heading">¶</a></h2>
<p>TreeProfiler <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand is the step that annotate input metadata to target tree. As a result, itwill generate the following output file:</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.nw</em>, newick format with annotated tree</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.ete</em>, ete format with annotated tree</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated_prop2type.txt</em>, config file where store the datatype of each annotated properties</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.tsv</em>,  metadata in tab-sarated values format with annotated and summarized internal nodes information.</p></li>
</ol>
<section id="annotate-metadata-into-tree">
<h3><a class="toc-backref" href="#id19" role="doc-backlink">Annotate metadata into tree</a><a class="headerlink" href="#annotate-metadata-into-tree" title="Link to this heading">¶</a></h3>
<p>In the following sub session we will describe the usage of following arguments in <code class="docutils literal notranslate"><span class="pre">annotate</span></code> step for metadata:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">-d,</span> <span class="pre">--metadata</span> <span class="pre">METADATA</span> <span class="pre">[METADATA</span> <span class="pre">...]</span></code></p></td>
<td><p>&lt;metadata.csv&gt; .csv, .tsv filename</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">-s,</span> <span class="pre">--metadata-s</span> <span class="pre">METADATA_SEP</span></code></p></td>
<td><p>Column separator of metadata table <code class="docutils literal notranslate"><span class="pre">[default:</span> <span class="pre">\t]</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--data-matrix</span> <span class="pre">DATA_MATRIX</span> <span class="pre">[DATA_MATRIX</span> <span class="pre">...]</span></code></p></td>
<td><p>&lt;datamatrix.csv&gt; .csv, .tsv. Numerical matrix data metadata table as array to tree, please do not provide column headers in this file, filename will become the property name in the tree.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--no-headers</span></code></p></td>
<td><p>Metadata table doesn’t contain columns name, namespace <code class="docutils literal notranslate"><span class="pre">col``+``index</span></code> will be assigned as the key of property such as <code class="docutils literal notranslate"><span class="pre">col1</span></code>.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--duplicate</span></code></p></td>
<td><p>Treeprofiler will aggregate duplicated metadata to a list as a property if metadata contains duplicated row.</p></td>
</tr>
</tbody>
</table>
<section id="basic-metadata-in-tsv-csv-format">
<h4><a class="toc-backref" href="#id20" role="doc-backlink">Basic metadata in TSV/CSV format</a><a class="headerlink" href="#basic-metadata-in-tsv-csv-format" title="Link to this heading">¶</a></h4>
<p>TreeProfiler allows users to input metadata in tsv/csv file by setting <code class="docutils literal notranslate"><span class="pre">--metadata</span> <span class="pre">&lt;filename.tsv|.csv&gt;</span></code>  and <code class="docutils literal notranslate"><span class="pre">-s</span> <span class="pre">&lt;seperator&gt;</span></code>. By default, the first column of metadata should be names of target tree leaves and metadata should contain column names for each column of metadata.</p>
<p>TreeProfiler allows user to annotate more than one metadata inputs to tree such as <code class="docutils literal notranslate"><span class="pre">--metadata</span> <span class="pre">table1.tsv</span> <span class="pre">table2.tsv</span></code>.</p>
<p>Check metadata</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>cd examples/basic_example0/
tree ./
./
├── boolean.tsv
├── categorical_duplicated.tsv
├── categorical.tsv
├── data.array
├── demo1.tree
├── numerical.tsv
└── show_tree_props.py

# check metadata structure
head categorical.tsv
name,categorical1
Taxa_0,A
Taxa_1,B
Taxa_2,B
Taxa_3,C
</pre></div>
</div>
<p>Run <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">## annotate tree with more than one metadata tsv, seperated by ``,``</span>
<span class="c1"># set the correct filename and seperator</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">s</span> <span class="p">,</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">.</span>
</pre></div>
</div>
<p>After annotation, treeprofiler will generate annotated tree</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="n">demo1</span><span class="o">*</span>
<span class="n">demo1_annotated</span><span class="o">.</span><span class="n">ete</span>  <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>  <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">tsv</span>  <span class="n">demo1_prop2type</span><span class="o">.</span><span class="n">txt</span>  <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span>
</pre></div>
</div>
<p>Now we can check annotated tree</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># show tree&#39;s properties</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>

<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;categorical1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;A--1||B--2||C--2&#39;</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span> <span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span> <span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A&#39;</span><span class="p">}</span>
</pre></div>
</div>
</section>
<section id="metadata-tsv-csv-as-a-numerical-data-matrix">
<h4><a class="toc-backref" href="#id21" role="doc-backlink">Metadata TSV/CSV as a numerical data matrix</a><a class="headerlink" href="#metadata-tsv-csv-as-a-numerical-data-matrix" title="Link to this heading">¶</a></h4>
<p>treeprofiler can handle the whole tsv/csv file as one property and annotate it to related leaves, by using <code class="docutils literal notranslate"><span class="pre">--data-matrix</span> <span class="pre">&lt;filename.tsv|.csv&gt;</span></code> It must be numerical data matrix and without headers. Once annotated the property of data-matrix will be named by the filename (see example below)</p>
<p>The difference between <code class="docutils literal notranslate"><span class="pre">--data-matrix</span></code> and <code class="docutils literal notranslate"><span class="pre">--metadata</span></code> is that the former sees the whole metadata file as a node property and stores the rows as an array in leaf nodes, and the latter sees each column from metadata as each single property of leaf nodes.</p>
<p>Using data array file <code class="docutils literal notranslate"><span class="pre">data.array</span></code> from the previous example</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># annotated data.array file to tree</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">data</span><span class="o">-</span><span class="n">matrix</span> <span class="n">data</span><span class="o">.</span><span class="n">array</span> \
<span class="o">-</span><span class="n">s</span> <span class="p">,</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">.</span>

<span class="c1"># data.array is stored as one property in tree node and value is stored as array</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;data.array_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;1.0244|-0.667|-1.7740000000000002|-0.8620000000000001|-0.6552&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array_max&#39;</span><span class="p">:</span> <span class="s1">&#39;3.671|1.937|4.362|1.585|2.746&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array_min&#39;</span><span class="p">:</span> <span class="s1">&#39;-2.591|-2.356|-4.825|-3.326|-2.479&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array_std&#39;</span><span class="p">:</span> <span class="s1">&#39;2.3121192529798287|1.5156064132880938|3.524138873540599|1.9937640783202009|1.906460531980665&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array_sum&#39;</span><span class="p">:</span> <span class="s1">&#39;5.122|-3.335|-8.870000000000001|-4.3100000000000005|-3.276&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;data.array&#39;</span><span class="p">:</span> <span class="s1">&#39;-2.591|1.937|-3.898|0.447|-1.349&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="metadata-without-column-names">
<h4><a class="toc-backref" href="#id22" role="doc-backlink">Metadata without column names</a><a class="headerlink" href="#metadata-without-column-names" title="Link to this heading">¶</a></h4>
<p>If metadata does not have headers, by setting <code class="docutils literal notranslate"><span class="pre">--no-headers</span></code> to set the metadata properly, therefore treeprofiler will name each column by <code class="docutils literal notranslate"><span class="pre">col</span></code> + <code class="docutils literal notranslate"><span class="pre">&lt;column</span> <span class="pre">number&gt;</span></code> as the property key in each leaf node, such as <code class="docutils literal notranslate"><span class="pre">col1</span></code>, <code class="docutils literal notranslate"><span class="pre">col2</span></code>, etc.</p>
<p>Example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># data.array doesn&#39;t have headers for each column</span>
<span class="n">head</span> <span class="n">data</span><span class="o">.</span><span class="n">array</span>
<span class="n">Taxa_0</span><span class="p">,</span><span class="o">-</span><span class="mf">2.591</span><span class="p">,</span><span class="mf">1.937</span><span class="p">,</span><span class="o">-</span><span class="mf">3.898</span><span class="p">,</span><span class="mf">0.447</span><span class="p">,</span><span class="o">-</span><span class="mf">1.349</span>
<span class="n">Taxa_1</span><span class="p">,</span><span class="mf">3.366</span><span class="p">,</span><span class="o">-</span><span class="mf">1.871</span><span class="p">,</span><span class="mf">4.362</span><span class="p">,</span><span class="mf">1.585</span><span class="p">,</span><span class="o">-</span><span class="mf">2.479</span>
<span class="n">Taxa_2</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="o">-</span><span class="mf">0.098</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="o">-</span><span class="mf">3.326</span><span class="p">,</span><span class="mf">2.746</span>
<span class="n">Taxa_3</span><span class="p">,</span><span class="mf">3.671</span><span class="p">,</span><span class="o">-</span><span class="mf">0.947</span><span class="p">,</span><span class="o">-</span><span class="mf">4.509</span><span class="p">,</span><span class="o">-</span><span class="mf">3.131</span><span class="p">,</span><span class="o">-</span><span class="mf">2.194</span>

<span class="c1"># need to add --no-headers flag to tell treeprofiler</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">data</span><span class="o">.</span><span class="n">array</span> \
<span class="o">-</span><span class="n">s</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">headers</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">.</span>

<span class="c1"># check properties</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;col1_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;1.92825&#39;</span><span class="p">,</span>
<span class="s1">&#39;col1_max&#39;</span><span class="p">:</span> <span class="s1">&#39;3.671&#39;</span><span class="p">,</span>
<span class="s1">&#39;col1_min&#39;</span><span class="p">:</span> <span class="s1">&#39;0.0&#39;</span><span class="p">,</span>
<span class="s1">&#39;col1_std&#39;</span><span class="p">:</span> <span class="s1">&#39;3.463526916666666&#39;</span><span class="p">,</span>
<span class="s1">&#39;col1_sum&#39;</span><span class="p">:</span> <span class="s1">&#39;7.713&#39;</span><span class="p">,</span>
<span class="s1">&#39;col2_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;-1.318&#39;</span><span class="p">,</span>
<span class="o">...</span><span class="p">}</span>

<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;col1&#39;</span><span class="p">:</span> <span class="s1">&#39;-2.591&#39;</span><span class="p">,</span>
<span class="s1">&#39;col2&#39;</span><span class="p">:</span> <span class="s1">&#39;1.937&#39;</span><span class="p">,</span>
<span class="s1">&#39;col3&#39;</span><span class="p">:</span> <span class="s1">&#39;-3.898&#39;</span><span class="p">,</span>
<span class="s1">&#39;col4&#39;</span><span class="p">:</span> <span class="s1">&#39;0.447&#39;</span><span class="p">,</span>
<span class="s1">&#39;col5&#39;</span><span class="p">:</span> <span class="s1">&#39;-1.349&#39;</span><span class="p">}</span>
</pre></div>
</div>
</section>
<section id="missing-value-detection">
<h4><a class="toc-backref" href="#id23" role="doc-backlink">Missing value detection</a><a class="headerlink" href="#missing-value-detection" title="Link to this heading">¶</a></h4>
<p>Metadata column which fullfills one of the following criterias will be consider as missing value:</p>
<ul class="simple">
<li><p>Entirely symbolic characters. Such as <code class="docutils literal notranslate"><span class="pre">+</span></code>, <code class="docutils literal notranslate"><span class="pre">-</span></code>, <code class="docutils literal notranslate"><span class="pre">~</span></code>, <code class="docutils literal notranslate"><span class="pre">.</span></code>, etc.</p></li>
<li><p>The exact strings <code class="docutils literal notranslate"><span class="pre">none</span></code>, <code class="docutils literal notranslate"><span class="pre">None</span></code>, <code class="docutils literal notranslate"><span class="pre">null</span></code>, <code class="docutils literal notranslate"><span class="pre">Null</span></code>, or <code class="docutils literal notranslate"><span class="pre">NaN</span></code>.</p></li>
<li><p>An empty string (zero characters).</p></li>
</ul>
<p>Missing value will replaced by string ‘NaN’ in the corresponding property.</p>
</section>
<section id="unmapped-tree-leaf-property-detection">
<h4><a class="toc-backref" href="#id24" role="doc-backlink">Unmapped tree leaf property detection</a><a class="headerlink" href="#unmapped-tree-leaf-property-detection" title="Link to this heading">¶</a></h4>
<p>If Metadata doesn’t cover input tree leaf, tree leaf will be unannotated.</p>
</section>
<section id="handling-duplicated-leaf-names-in-metadata">
<h4><a class="toc-backref" href="#id25" role="doc-backlink">Handling duplicated leaf names in metadata</a><a class="headerlink" href="#handling-duplicated-leaf-names-in-metadata" title="Link to this heading">¶</a></h4>
<p>In general, treeprofiler expects each row of metadata corresponding to one leaf, such as</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span>
<span class="c1">#name,categorical1</span>
<span class="n">Taxa_0</span><span class="p">,</span><span class="n">A</span>
<span class="n">Taxa_1</span><span class="p">,</span><span class="n">B</span>
<span class="n">Taxa_2</span><span class="p">,</span><span class="n">B</span>
<span class="n">Taxa_3</span><span class="p">,</span><span class="n">C</span>
<span class="n">Taxa_4</span><span class="p">,</span><span class="n">C</span>
</pre></div>
</div>
<p>Although treeprofiler can handle metadata with rows with duplicated leafnames such as</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">head</span> <span class="n">categorical_duplicated</span><span class="o">.</span><span class="n">tsv</span>
<span class="c1">#name,categorical1</span>
<span class="n">Taxa_0</span><span class="p">,</span><span class="n">A</span>
<span class="n">Taxa_0</span><span class="p">,</span><span class="n">B</span>
<span class="n">Taxa_2</span><span class="p">,</span><span class="n">B</span>
<span class="n">Taxa_2</span><span class="p">,</span><span class="n">C</span>
<span class="n">Taxa_3</span><span class="p">,</span><span class="n">C</span>
<span class="n">Taxa_3</span><span class="p">,</span><span class="n">A</span>
<span class="n">Taxa_4</span><span class="p">,</span><span class="n">C</span>
</pre></div>
</div>
<p>In order to do so, users need to add <code class="docutils literal notranslate"><span class="pre">--duplicate</span></code> , by doing so, metadata from the same leaf will be aggregate into the same column. Such as the Taxa_0 from the above table, at the end value <code class="docutils literal notranslate"><span class="pre">A</span></code> and <code class="docutils literal notranslate"><span class="pre">B</span></code> will be both annotated to property <code class="docutils literal notranslate"><span class="pre">categorical1</span></code> (see above demo).
<strong>If not, treeprofielr will take one the first row of metadata that appear as the metadata for related leaf!</strong></p>
<p>example</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">-</span><span class="n">d</span> <span class="n">categorical_duplicated</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">s</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">duplicate</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">.</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;categorical1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;A--2||B--2||C--3&#39;</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span> <span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span> <span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A|B&#39;</span><span class="p">}</span>
</pre></div>
</div>
</section>
<section id="specifying-data-types-for-metadata-columns">
<h4><a class="toc-backref" href="#id26" role="doc-backlink">Specifying data types for metadata columns</a><a class="headerlink" href="#specifying-data-types-for-metadata-columns" title="Link to this heading">¶</a></h4>
<p>Although TreeProfiler can automatically detect datatype of each column, users still can determine the datatype using the following arguments using:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--text-prop</span> <span class="pre">TEXT_PROP</span> <span class="pre">[TEXT_PROP</span> <span class="pre">...]</span></code></p></td>
<td><p>names of columns which need to be read as categorical data</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--multiple-text-prop</span> <span class="pre">MULTIPLE_TEXT_PROP</span> <span class="pre">[MULTIPLE_TEXT_PROP</span> <span class="pre">...]</span></code></p></td>
<td><p>names of columns which need to be read as categorical data containing more than one value and separated by <cite>,</cite> such as <a class="reference external" href="GO:0000003,GO:0000902,GO:0000904,GO:0003006">GO:0000003,GO:0000902,GO:0000904,GO:0003006</a></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--num-prop</span> <span class="pre">NUM_PROP</span> <span class="pre">[NUM_PROP</span> <span class="pre">...]</span></code></p></td>
<td><p>names of columns which need to be read as numerical data</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--bool-prop</span> <span class="pre">BOOL_PROP</span> <span class="pre">[BOOL_PROP</span> <span class="pre">...]</span></code></p></td>
<td><p>names of columns which need to be read as boolean data</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--text-prop-idx</span> <span class="pre">TEXT_PROP_IDX</span> <span class="pre">[TEXT_PROP_IDX</span> <span class="pre">...]</span></code></p></td>
<td><p>1 2 3 or [1-5] index of columns which need to be read as categorical data</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--num-prop-idx</span> <span class="pre">NUM_PROP_IDX</span> <span class="pre">[NUM_PROP_IDX</span> <span class="pre">...]</span></code></p></td>
<td><p>1 2 3 or [1-5] index columns which need to be read as numerical data</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--bool-prop-idx</span> <span class="pre">BOOL_PROP_IDX</span> <span class="pre">[BOOL_PROP_IDX</span> <span class="pre">...]</span></code></p></td>
<td><p>1 2 3 or [1-5] index columns which need to be read as boolean data</p></td>
</tr>
</tbody>
</table>
</section>
<section id="annotate-metadata-to-internal-nodes">
<h4><a class="toc-backref" href="#id27" role="doc-backlink">Annotate metadata to internal nodes</a><a class="headerlink" href="#annotate-metadata-to-internal-nodes" title="Link to this heading">¶</a></h4>
<p>At the above example, we only mapped metadata to leaf nodes, in this example, we will also profile <strong>internal nodes</strong> annotation and analysis of their children nodes. Argument that in related to summary methods are:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Applied datatype</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Summarized properties Internal node</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--num-stat</span> <span class="pre">{all,sum,avg,max,min,std,none}</span></code></p></td>
<td><p>numerical data matrix</p></td>
<td><p>Descriptive Statistic (average, sum, max, min, standard deviation)</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_avg</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_sum</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_max</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_min</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_std</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--counter-stat</span> <span class="pre">{raw,relative,none}</span></code></p></td>
<td><p>str
boolean
list</p></td>
<td><p>Raw/Relative Counter</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_counter</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--num-stat</span> <span class="pre">{all,sum,avg,max,min,std,none}</span></code></p></td>
<td><p>float
int</p></td>
<td><p>Descriptive Statistic (average, sum, max, min, standard deviation)</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_avg</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_sum</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_max</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_min</span></code>
<code class="docutils literal notranslate"><span class="pre">&lt;prop</span> <span class="pre">name&gt;_std</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--column-summary-method</span> <span class="pre">COLUMN_SUMMARY_METHOD</span> <span class="pre">[COLUMN_SUMMARY_METHOD</span> <span class="pre">...]</span></code></p></td>
<td><p>all</p></td>
<td><p>Specify summary method for individual columns in the format ColumnName=Method, such as <code class="docutils literal notranslate"><span class="pre">--column-summary-method</span> <span class="pre">sample1=none</span> <span class="pre">sample2=avg</span> <span class="pre">random_type=relative</span> <span class="pre">alignment=none</span></code></p></td>
<td></td>
</tr>
</tbody>
</table>
<p>TreeProfiler can infer automatically the datatype of each column in your metadata, including</p>
<ul class="simple">
<li><p><cite>list</cite> (seperate by <cite>,</cite> )</p></li>
<li><p><cite>string</cite> (categorcial data)</p></li>
<li><p><cite>numerical</cite> (numerical data, float or integer)</p></li>
<li><p><cite>booleans</cite></p></li>
</ul>
<p>Internal node will summurize children nodes information according to their datatypes.</p>
<p>demo tree</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>      ╭╴A
╴root╶┤
      │   ╭╴B
      ╰╴D╶┤
          ╰╴C
</pre></div>
</div>
<p>demo metadata</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>#name</p></th>
<th class="head"><p>text_property</p></th>
<th class="head"><p>multiple_text_property</p></th>
<th class="head"><p>numerical_property</p></th>
<th class="head"><p>bool_property</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>A</p></td>
<td><p>vowel</p></td>
<td><p>a,b,c</p></td>
<td><p>10</p></td>
<td><p>True</p></td>
</tr>
<tr class="row-odd"><td><p>B</p></td>
<td><p>consonant</p></td>
<td><p>b,c,d</p></td>
<td><p>4</p></td>
<td><p>False</p></td>
</tr>
<tr class="row-even"><td><p>C</p></td>
<td><p>consonant</p></td>
<td><p>c,d,e</p></td>
<td><p>9</p></td>
<td><p>True</p></td>
</tr>
</tbody>
</table>
<p>TreeProfiler will infer the datatypes of above metadata and adopt different summary methods:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><ul class="simple">
<li></li>
</ul>
</th>
<th class="head"><p>text_property</p></th>
<th class="head"><p>multiple_text_property</p></th>
<th class="head"><p>numerical_property</p></th>
<th class="head"><p>bool_property</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>datatype</p></td>
<td><p>string</p></td>
<td><p>list</p></td>
<td><p>float</p></td>
<td><p>bool</p></td>
</tr>
<tr class="row-odd"><td><p>method</p></td>
<td><p>counter</p></td>
<td><p>counter</p></td>
<td><p>average, sum, max, min, standard deviation</p></td>
<td><p>counter</p></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li><p>Categorical</p></li>
</ol>
<p>boolean and text properties (categorical data) of leaf nodes will be summarized as counters in internal nodes, currently users can choose using <code class="docutils literal notranslate"><span class="pre">raw</span></code> (default), <code class="docutils literal notranslate"><span class="pre">relative</span></code> or <code class="docutils literal notranslate"><span class="pre">none</span></code> for counter. Users can use <code class="docutils literal notranslate"><span class="pre">--counter-stat</span> <span class="pre">{raw,relative,none}</span></code> to choose the counter, it will automatically apply to all categorical properties.</p>
<p>After annotation, internal nodes will be summarized. If property was summarize with <code class="docutils literal notranslate"><span class="pre">counter</span></code>, in internal node will be named as <code class="docutils literal notranslate"><span class="pre">&lt;property_name&gt;_counter</span></code></p>
<p>Users can choose either counter is raw or relative count by using <code class="docutils literal notranslate"><span class="pre">--counter-stat</span></code></p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>internal_node properties</p></th>
<th class="head"><p>statistic method</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;prop name&gt;_counter</p></td>
<td><p>raw(default), relative</p></td>
</tr>
</tbody>
</table>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>internal_node</p></th>
<th class="head"><p>text_property_counter</p></th>
<th class="head"><p>multiple_text_property_counter</p></th>
<th class="head"><p>bool_property_counter</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>D</p></td>
<td><p>consonant–2</p></td>
<td><p>b–1||c–2||d–2||e–1</p></td>
<td><p>True–1||False–1</p></td>
</tr>
<tr class="row-odd"><td><p>root</p></td>
<td><p>vowel–1||consonant–2</p></td>
<td><p>a–2||b–2||c–3||d–2||e–1</p></td>
<td><p>True–2||False–1</p></td>
</tr>
</tbody>
</table>
<p>Example</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># raw counter (default)</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">counter</span><span class="o">-</span><span class="n">stat</span> <span class="n">raw</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;categorical1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;A--1||B--2||C--2&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A&#39;</span>
<span class="p">}</span>

<span class="c1">#relative counter to calculate the percentage</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">counter</span><span class="o">-</span><span class="n">stat</span> <span class="n">relative</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;categorical1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;A--0.20||B--0.40||C--0.40&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A&#39;</span>
<span class="p">}</span>

<span class="c1">#set to none</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">counter</span><span class="o">-</span><span class="n">stat</span> <span class="n">none</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span> <span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span> <span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A&#39;</span><span class="p">}</span>
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Numerical</p></li>
</ol>
<p>By default, numerical feature will be calculated all the descriptive statistic, but users can choose specific one to be calculated by using <code class="docutils literal notranslate"><span class="pre">--num-stat</span> <span class="pre">{all,</span> <span class="pre">sum,</span> <span class="pre">avg,</span> <span class="pre">max,</span> <span class="pre">min,</span> <span class="pre">std,</span> <span class="pre">none}</span></code>. <code class="docutils literal notranslate"><span class="pre">all</span></code> (default) means it will conduct all the statistic. <code class="docutils literal notranslate"><span class="pre">none</span></code> means annotation will only conduct in leaf nodes.</p>
<p>If property was numerical data, in internal node will be named as</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>internal_node properties</p></th>
<th class="head"><p>statistic method</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>&lt;prop name&gt;_avg</p></td>
<td><p>average</p></td>
</tr>
<tr class="row-odd"><td><p>&lt;prop name&gt;_sum</p></td>
<td><p>sum</p></td>
</tr>
<tr class="row-even"><td><p>&lt;prop name&gt;_max</p></td>
<td><p>maximum</p></td>
</tr>
<tr class="row-odd"><td><p>&lt;prop name&gt;_min</p></td>
<td><p>minimum</p></td>
</tr>
<tr class="row-even"><td><p>&lt;prop name&gt;_std</p></td>
<td><p>standard deviation</p></td>
</tr>
</tbody>
</table>
<p>Noticed that <code class="docutils literal notranslate"><span class="pre">--num-stat</span></code> will also work on <code class="docutils literal notranslate"><span class="pre">--data-matrix</span></code> data.</p>
<p>In our demo, it would be:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>internal_node</p></th>
<th class="head"><p>numerical_property_avg</p></th>
<th class="head"><p>numerical_property_sum</p></th>
<th class="head"><p>numerical_property_max</p></th>
<th class="head"><p>numerical_property_max</p></th>
<th class="head"><p>numerical_property_max</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>D</p></td>
<td><p>6.5</p></td>
<td><p>13</p></td>
<td><p>9</p></td>
<td><p>4</p></td>
<td><p>2.5</p></td>
</tr>
<tr class="row-odd"><td><p>root</p></td>
<td><p>7.67</p></td>
<td><p>23</p></td>
<td><p>10</p></td>
<td><p>4</p></td>
<td><p>2.32</p></td>
</tr>
</tbody>
</table>
<p>Example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># conduct all statistic (by default)</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">numerical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">num</span><span class="o">-</span><span class="n">stat</span> <span class="nb">all</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;0.5384554640742852&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_max&#39;</span><span class="p">:</span> <span class="s1">&#39;0.7817176831389784&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_min&#39;</span><span class="p">:</span> <span class="s1">&#39;0.3276816717486982&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_std&#39;</span><span class="p">:</span> <span class="s1">&#39;0.028430041000376213&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_sum&#39;</span><span class="p">:</span> <span class="s1">&#39;2.692277320371426&#39;</span><span class="p">,</span>
<span class="o">....</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;random_column1&#39;</span><span class="p">:</span> <span class="s1">&#39;0.45303222603186877&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column2&#39;</span><span class="p">:</span> <span class="s1">&#39;1.9801547427961053&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3&#39;</span><span class="p">:</span> <span class="s1">&#39;43.0&#39;</span><span class="p">}</span>

<span class="c1"># conduct only average</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">numerical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">num</span><span class="o">-</span><span class="n">stat</span> <span class="n">avg</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;0.5384554640742852&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column2_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;0.12655333321138568&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;52.2&#39;</span>
<span class="p">}</span>

<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;random_column1&#39;</span><span class="p">:</span> <span class="s1">&#39;0.45303222603186877&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column2&#39;</span><span class="p">:</span> <span class="s1">&#39;1.9801547427961053&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3&#39;</span><span class="p">:</span> <span class="s1">&#39;43.0&#39;</span><span class="p">}</span>


<span class="c1"># conduct none statistic</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">numerical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">num</span><span class="o">-</span><span class="n">stat</span> <span class="n">none</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span><span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;random_column1&#39;</span><span class="p">:</span> <span class="s1">&#39;0.45303222603186877&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column2&#39;</span><span class="p">:</span> <span class="s1">&#39;1.9801547427961053&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3&#39;</span><span class="p">:</span> <span class="s1">&#39;43.0&#39;</span>
<span class="p">}</span>

<span class="c1"># data matrix is also effected by --num-stat setting</span>

<span class="c1"># only average</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">data</span><span class="o">-</span><span class="n">matrix</span> <span class="n">data</span><span class="o">.</span><span class="n">array</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">num</span><span class="o">-</span><span class="n">stat</span> <span class="n">avg</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;data.array_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;1.0244|-0.667|-1.7740000000000002|-0.8620000000000001|-0.6552&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;data.array&#39;</span><span class="p">:</span> <span class="s1">&#39;-2.591|1.937|-3.898|0.447|-1.349&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
<section id="customizing-summary-methods-for-different-columns">
<h5><a class="toc-backref" href="#id28" role="doc-backlink">Customizing summary methods for different columns</a><a class="headerlink" href="#customizing-summary-methods-for-different-columns" title="Link to this heading">¶</a></h5>
<p>Using <code class="docutils literal notranslate"><span class="pre">--column-summary-method</span></code>  can specify the summary method of each properties, simply add <code class="docutils literal notranslate"><span class="pre">&lt;property</span> <span class="pre">name&gt;=&lt;summary</span> <span class="pre">method&gt;</span></code> . For categorical data, options are <code class="docutils literal notranslate"><span class="pre">{raw,relative,none}</span></code>; for numerical data, options are  <code class="docutils literal notranslate"><span class="pre">{all,</span> <span class="pre">sum,</span> <span class="pre">avg,</span> <span class="pre">max,</span> <span class="pre">min,</span> <span class="pre">std,</span> <span class="pre">none}</span></code> .</p>
<p>such as <code class="docutils literal notranslate"><span class="pre">--column-summary-method</span> <span class="pre">sample1=none</span> <span class="pre">sample2=avg</span> <span class="pre">random_type=relative</span> <span class="pre">alignment=none</span></code></p>
<p>Noted that <code class="docutils literal notranslate"><span class="pre">--data-matrix</span></code> can be effected by <code class="docutils literal notranslate"><span class="pre">--column-summary-method</span> <span class="pre">``</span> <span class="pre">setting,</span> <span class="pre">in</span> <span class="pre">this</span> <span class="pre">case</span> <span class="pre">filename</span> <span class="pre">of</span> <span class="pre">the</span> <span class="pre">data</span> <span class="pre">matrix</span> <span class="pre">is</span> <span class="pre">property</span> <span class="pre">name,</span> <span class="pre">such</span> <span class="pre">as</span> <span class="pre">``--data-matrix</span> <span class="pre">file.tsv</span> <span class="pre">--column-summary-method</span> <span class="pre">file.tsv=avg</span></code></p>
<p>example:
here we use three different metadata: <code class="docutils literal notranslate"><span class="pre">categorical.tsv</span></code>, <code class="docutils literal notranslate"><span class="pre">numerical.tsv</span></code> and <code class="docutils literal notranslate"><span class="pre">data</span> <span class="pre">matrix</span></code></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># cusomtize different summary methods for different column/property</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo1</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">categorical</span><span class="o">.</span><span class="n">tsv</span> <span class="n">numerical</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">--</span><span class="n">data</span><span class="o">-</span><span class="n">matrix</span> <span class="n">data</span><span class="o">.</span><span class="n">array</span> \
<span class="o">-</span><span class="n">sep</span> <span class="p">,</span> \
<span class="o">--</span><span class="n">column</span><span class="o">-</span><span class="n">summary</span><span class="o">-</span><span class="n">method</span> \
<span class="n">categorical1</span><span class="o">=</span><span class="n">relative</span> \
<span class="n">random_column1</span><span class="o">=</span><span class="nb">all</span> \
<span class="n">random_column2</span><span class="o">=</span><span class="n">none</span> \
<span class="n">random_column3</span><span class="o">=</span><span class="nb">sum</span> \
<span class="n">data</span><span class="o">.</span><span class="n">array</span><span class="o">=</span><span class="n">avg</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo1_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;categorical1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;A--0.20||B--0.40||C--0.40&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;0.5384554640742852&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_max&#39;</span><span class="p">:</span> <span class="s1">&#39;0.7817176831389784&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_min&#39;</span><span class="p">:</span> <span class="s1">&#39;0.3276816717486982&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_std&#39;</span><span class="p">:</span> <span class="s1">&#39;0.028430041000376213&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1_sum&#39;</span><span class="p">:</span> <span class="s1">&#39;2.692277320371426&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3_sum&#39;</span><span class="p">:</span> <span class="s1">&#39;261.0&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array_avg&#39;</span><span class="p">:</span> <span class="s1">&#39;1.0244|-0.667|-1.7740000000000002|-0.8620000000000001|-0.6552&#39;</span>
<span class="p">}</span>
<span class="n">target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_0&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.190563</span><span class="p">,</span>
<span class="s1">&#39;categorical1&#39;</span><span class="p">:</span> <span class="s1">&#39;A&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column1&#39;</span><span class="p">:</span> <span class="s1">&#39;0.45303222603186877&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column2&#39;</span><span class="p">:</span> <span class="s1">&#39;1.9801547427961053&#39;</span><span class="p">,</span>
<span class="s1">&#39;random_column3&#39;</span><span class="p">:</span> <span class="s1">&#39;43.0&#39;</span><span class="p">,</span>
<span class="s1">&#39;data.array&#39;</span><span class="p">:</span> <span class="s1">&#39;-2.591|1.937|-3.898|0.447|-1.349&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
</section>
<section id="taxonomic-annotation">
<h3><a class="toc-backref" href="#id29" role="doc-backlink">Taxonomic annotation</a><a class="headerlink" href="#taxonomic-annotation" title="Link to this heading">¶</a></h3>
<p>Treeprofiler annotate tree node with target taxonomy, you can either use [GTDB](<a class="reference external" href="https://gtdb.ecogenomic.org/">https://gtdb.ecogenomic.org/</a>) or [NCBI](<a class="reference external" href="https://www.ncbi.nlm.nih.gov/">https://www.ncbi.nlm.nih.gov/</a>) taxonomic database, such as following commands</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">TAXON_COLUMN</span></code></p></td>
<td><p>Choose the column in metadata which represents taxon for activating the taxonomic annotation. Default is the first column, which should be the column of leaf_name.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--taxadb</span> <span class="pre">{NCBI,GTDB,customdb}</span></code></p></td>
<td><p>NCBI or GTDB, choose the Taxonomic Database for annotation.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--taxon-delimiter</span> <span class="pre">TAXON_DELIMITER</span></code></p></td>
<td><p>Delimiter of taxa columns. <code class="docutils literal notranslate"><span class="pre">[default:</span> <span class="pre">None]</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--taxa-field</span> <span class="pre">TAXA_FIELD</span></code></p></td>
<td><p>Field of taxa name after delimiter. <code class="docutils literal notranslate"><span class="pre">[default:</span> <span class="pre">0]</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--taxa-dump</span> <span class="pre">TAXA_DUMP</span></code></p></td>
<td><p>Path to taxonomic database dump file for a specific version, such as GTDB taxadump (<a class="reference external" href="https://github.com/etetoolkit/ete-data/raw/main/gtdb_taxonomy/gtdblatest/gtdb_latest_dump.tar.gz">https://github.com/etetoolkit/ete-data/raw/main/gtdb_taxonomy/gtdblatest/gtdb_latest_dump.tar.gz</a>) or NCBI taxadump (<a class="reference external" href="https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz">https://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz</a>).</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--gtdb-version</span> <span class="pre">{95,202,207,214,220}</span></code></p></td>
<td><p>GTDB version for taxonomic annotation, such as 220. If it is not provided, the latest version will be used.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--ignore-unclassified</span></code></p></td>
<td><p>Ignore unclassified taxa in taxonomic annotation.</p></td>
</tr>
</tbody>
</table>
<p>In this part we will demostrate the usage of taxonomic annotation in examples of <code class="docutils literal notranslate"><span class="pre">examples/taxonomy_example</span></code></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="n">examples</span><span class="o">/</span><span class="n">taxonomy_example</span>
<span class="n">ls</span> <span class="o">./</span>
<span class="n">demo3</span><span class="o">.</span><span class="n">tree</span>  <span class="n">demo4</span><span class="o">.</span><span class="n">tree</span>  <span class="n">gtdb202dump</span><span class="o">.</span><span class="n">tar</span><span class="o">.</span><span class="n">gz</span>  <span class="n">missing_gtdb_v202</span><span class="o">.</span><span class="n">tree</span>  <span class="n">ncbi</span><span class="o">.</span><span class="n">tree</span>
<span class="n">demo3</span><span class="o">.</span><span class="n">tsv</span>   <span class="n">demo4</span><span class="o">.</span><span class="n">tsv</span>   <span class="n">gtdb_v202</span><span class="o">.</span><span class="n">tree</span>      <span class="n">missing_ncbi</span><span class="o">.</span><span class="n">tree</span>       <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
<section id="using-different-taxonomic-databases-from-gtdb-ncbi">
<h4><a class="toc-backref" href="#id30" role="doc-backlink">Using different taxonomic databases from GTDB/NCBI</a><a class="headerlink" href="#using-different-taxonomic-databases-from-gtdb-ncbi" title="Link to this heading">¶</a></h4>
<p>To start taxonomic annotation, using <code class="docutils literal notranslate"><span class="pre">--taxon-column</span></code> and <code class="docutils literal notranslate"><span class="pre">--taxadb</span></code> to locate where is the taxon and which taxonomic databases to be used. If taxon is leaf name, then using <code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">name</span></code>. Otherwise <code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">&lt;prop_name&gt;</span></code> which refers to the column in the metadata.</p>
<p>Examples in NCBI taxonomic database</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># check example tree</span>
<span class="n">cat</span> <span class="n">ncbi</span><span class="o">.</span><span class="n">tree</span>
<span class="p">((</span><span class="mi">9606</span><span class="p">,</span> <span class="mi">9598</span><span class="p">),</span> <span class="mi">10090</span><span class="p">);</span>

<span class="c1"># run taxonomic annotation and locate taxon column in leaf name</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">ncbi</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">ncbi</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="c1"># check annotation results</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">ncbi_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;no rank-cellular organisms|superkingdom-Eukaryota|clade-Eumetazoa|phylum-Chordata|superclass-Sarcopterygii|kingdom-Metazoa|class-Mammalia|subphylum-Craniata|superorder-Euarchontoglires&#39;</span><span class="p">,</span>
<span class="s1">&#39;lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;1|131567|2759|33154|33208|6072|33213|33511|7711|89593|7742|7776|117570|117571|8287|1338369|32523|32524|40674|32525|9347|1437010|314146&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|Eukaryota|Eumetazoa|Chordata|Vertebrata|Gnathostomata|Sarcopterygii|Eutheria|Tetrapoda|Amniota|Theria|Opisthokonta|Metazoa|Bilateria|Deuterostomia|Mammalia|Craniata|Teleostomi|Euteleostomi|cellular organisms|Euarchontoglires|Dipnotetrapodomorpha|Boreoeutheria&#39;</span><span class="p">,</span> <span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;superorder&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Euarchontoglires&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;10090|9606|9598&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;314146&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;9606&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span>
<span class="s1">&#39;Homo sapiens&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;no rank-cellular organisms|superkingdom-Eukaryota|clade-Eumetazoa|phylum-Chordata|superclass-Sarcopterygii|order-Primates|parvorder-Catarrhini|family-Hominidae|genus-Homo|species-Homo sapiens|kingdom-Metazoa|class-Mammalia|subphylum-Craniata|subfamily-Homininae|superorder-Euarchontoglires|infraorder-Simiiformes|superfamily-Hominoidea|suborder-Haplorrhini&#39;</span><span class="p">,</span>
<span class="s1">&#39;lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;1|131567|2759|33154|33208|6072|33213|33511|7711|89593|7742|7776|117570|117571|8287|1338369|32523|32524|40674|32525|9347|1437010|314146|9443|376913|314293|9526|314295|9604|207598|9605|9606&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|Eukaryota|Eumetazoa|Chordata|Vertebrata|Gnathostomata|Sarcopterygii|Eutheria|Primates|Catarrhini|Hominidae|Homo|Homo sapiens|Tetrapoda|Amniota|Theria|Opisthokonta|Metazoa|Bilateria|Deuterostomia|Mammalia|Craniata|Teleostomi|Euteleostomi|cellular organisms|Homininae|Euarchontoglires|Simiiformes|Hominoidea|Haplorrhini|Dipnotetrapodomorpha|Boreoeutheria&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;species&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;Homo sapiens&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;9606&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;9606&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>Examples in GTDB taxonomic database</p>
<p>For gtdb taxa, users can choose <code class="docutils literal notranslate"><span class="pre">--gtdb-version</span> <span class="pre">{95,202,207,214,220}</span></code> to select certain version, if not, latest gtdb db will be used.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># check example tree</span>
<span class="n">cat</span> <span class="n">gtdb_v202</span><span class="o">.</span><span class="n">tree</span>
<span class="p">(</span><span class="n">GB_GCA_011358815</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">,(</span><span class="n">RS_GCF_000019605</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">,(</span><span class="n">RS_GCF_003948265</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="n">GB_GCA_003344655</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">):</span><span class="mf">0.5</span><span class="p">):</span><span class="mf">0.5</span><span class="p">);</span>

<span class="c1"># default using latest version, in this case on tree from version 202, it should go empty</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">gtdb_v202</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">gtdb</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">gtdb_v202_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span> <span class="s1">&#39;lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;Unknown&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;None&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;RS_GCF_000019605.1|RS_GCF_003948265.1|GB_GCA_011358815.1|GB_GCA_003344655.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;None&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;Unknown&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span>
<span class="p">}</span>

<span class="c1">#annotate tree using the proper version of GTDB</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">gtdb_v202</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">gtdb</span> \
<span class="o">--</span><span class="n">gtdb</span><span class="o">-</span><span class="n">version</span> <span class="mi">202</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="c1"># now it&#39;s correctly annotated</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">gtdb_v202_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;superkingdom-d__Archaea|phylum-p__Thermoproteota|class-c__Korarchaeia|order-o__Korarchaeales|family-f__Korarchaeaceae|genus-g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;1|2|79|2172|2173|2174|2175&#39;</span><span class="p">,</span> <span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|d__Archaea|p__Thermoproteota|c__Korarchaeia|o__Korarchaeales|f__Korarchaeaceae|g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;genus&#39;</span><span class="p">,</span> <span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;RS_GCF_003948265.1|GB_GCA_011358815.1|RS_GCF_000019605.1|GB_GCA_003344655.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;superkingdom-d__Archaea|phylum-p__Thermoproteota|class-c__Korarchaeia|order-o__Korarchaeales|family-f__Korarchaeaceae|genus-g__Korarchaeum|species-s__Korarchaeum cryptofilum|subspecies-s__Korarchaeum cryptofilum&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|d__Archaea|p__Thermoproteota|c__Korarchaeia|o__Korarchaeales|f__Korarchaeaceae|g__Korarchaeum|s__Korarchaeum cryptofilum|GB_GCA_011358815.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;subspecies&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;s__Korarchaeum cryptofilum&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;GB_GCA_011358815.1&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="identifying-taxon-names-in-different-metadata-fields-columns">
<h4><a class="toc-backref" href="#id31" role="doc-backlink">Identifying taxon names in different metadata fields/columns</a><a class="headerlink" href="#identifying-taxon-names-in-different-metadata-fields-columns" title="Link to this heading">¶</a></h4>
<p>When Taxon properties are embeded in different column or field in metadata, treeprofiler provides <code class="docutils literal notranslate"><span class="pre">--taxon-column</span></code>, <code class="docutils literal notranslate"><span class="pre">--taxon-delimiter</span></code> and <code class="docutils literal notranslate"><span class="pre">--taxa-field</span></code> to identify taxon term in order to process taxonomic annotation sucessfully. Here is summary of different cases with corresponding setting.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>metadata</p></th>
<th class="head"><p>taxon to be identified</p></th>
<th class="head"><p>command line setting</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">col1\n9598</span> <span class="pre">wt</span></code></p></td>
<td><p>9598</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">default</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">col1\n7739.XP_002609184.1</span> <span class="pre">wt</span></code></p></td>
<td><p>7739</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">&lt;default&gt;</span> <span class="pre">--taxon-delimiter</span> <span class="pre">.</span> <span class="pre">--taxa-field</span> <span class="pre">0</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">ncbi_id\nleaf_A</span> <span class="pre">7739</span></code></p></td>
<td><p>7739</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">ncbi_id</span> <span class="pre">--taxon-delimiter</span> <span class="pre">&lt;default&gt;</span> <span class="pre">--taxa-field</span> <span class="pre">&lt;default&gt;</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">ncbi_id\nleaf_A</span> <span class="pre">7739.XP_002609184.1</span></code></p></td>
<td><p>7739</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">ncbi_id</span> <span class="pre">--taxon-delimiter</span> <span class="pre">.</span> <span class="pre">--taxa-field</span> <span class="pre">0</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">col1\nRS_GCF_001560035.1</span> <span class="pre">wt</span></code></p></td>
<td><p>RS_GCF_001560035.1</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">default</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">#leafname</span> <span class="pre">gtdb_id\nleaf_A</span> <span class="pre">d__Archaea;p__Asgardarchaeota;c__Heimdallarchaeia;o__UBA460;f__Kariarchaeaceae;g__LC-2;s__LC-2</span> <span class="pre">sp001940725</span></code></p></td>
<td><p>s__LC-2 sp001940725</p></td>
<td><p><code class="docutils literal notranslate"><span class="pre">--taxon-column</span> <span class="pre">gtdb_id</span> <span class="pre">--taxon-delimiter</span> <span class="pre">;</span> <span class="pre">--taxa-field</span> <span class="pre">-1</span></code></p></td>
</tr>
</tbody>
</table>
<p>Example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># check example tree and metadata</span>
<span class="n">cat</span> <span class="n">demo3</span><span class="o">.</span><span class="n">tree</span>
<span class="p">(</span><span class="n">Taxa_2</span><span class="p">:</span><span class="mf">0.471596</span><span class="p">,((</span><span class="n">Taxa_0</span><span class="p">:</span><span class="mf">0.767844</span><span class="p">,</span><span class="n">Taxa_1</span><span class="p">:</span><span class="mf">0.792161</span><span class="p">)</span><span class="mf">0.313833</span><span class="p">:</span><span class="mf">0.684109</span><span class="p">,</span><span class="n">Taxa_3</span><span class="p">:</span><span class="mf">0.805286</span><span class="p">):</span><span class="mf">0.188666</span><span class="p">);</span>

<span class="n">cat</span> <span class="n">demo3</span><span class="o">.</span><span class="n">tsv</span>
<span class="c1">#name gtdb_taxid</span>
<span class="n">Taxa_0</span>        <span class="n">GB_GCA_011358815</span><span class="mf">.1</span><span class="nd">@sample1</span>
<span class="n">Taxa_1</span>        <span class="n">RS_GCF_000019605</span><span class="mf">.1</span><span class="nd">@sample2</span>
<span class="n">Taxa_2</span>        <span class="n">RS_GCF_003948265</span><span class="mf">.1</span><span class="nd">@sample3</span>
<span class="n">Taxa_3</span>        <span class="n">GB_GCA_003344655</span><span class="mf">.1</span><span class="nd">@sample4</span>

<span class="c1"># therefore, locate taxa id correctly</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo3</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">-</span><span class="n">d</span> <span class="n">demo3</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">gtdb_taxid</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">gtdb</span> \
<span class="o">--</span><span class="n">gtdb</span><span class="o">-</span><span class="n">version</span> <span class="mi">202</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">delimiter</span> <span class="o">@</span> \
<span class="o">--</span><span class="n">taxa</span><span class="o">-</span><span class="n">field</span> <span class="mi">0</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo3_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;superkingdom-d__Archaea|phylum-p__Thermoproteota|class-c__Korarchaeia|order-o__Korarchaeales|family-f__Korarchaeaceae|genus-g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|d__Archaea|p__Thermoproteota|c__Korarchaeia|o__Korarchaeales|f__Korarchaeaceae|g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;genus&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_3|Taxa_0|Taxa_1|Taxa_2&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_2&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.471596</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;gtdb_taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;RS_GCF_003948265.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;superkingdom-d__Archaea|phylum-p__Thermoproteota|class-c__Korarchaeia|order-o__Korarchaeales|family-f__Korarchaeaceae|genus-g__Korarchaeum|species-s__Korarchaeum cryptofilum|subspecies-s__Korarchaeum cryptofilum&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|d__Archaea|p__Thermoproteota|c__Korarchaeia|o__Korarchaeales|f__Korarchaeaceae|g__Korarchaeum|s__Korarchaeum cryptofilum|RS_GCF_003948265.1&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;subspecies&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;s__Korarchaeum cryptofilum&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_2&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;RS_GCF_003948265.1&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="ignoring-unclassified-taxonomic-annotations">
<h4><a class="toc-backref" href="#id32" role="doc-backlink">Ignoring unclassified taxonomic annotations</a><a class="headerlink" href="#ignoring-unclassified-taxonomic-annotations" title="Link to this heading">¶</a></h4>
<p>Taxonomic annotation will annotate the internal nodes based on the taxa of leaf nodes, but if leaf node has unknown taxonomic information, the internal nodes will return unknown annotation. Using <code class="docutils literal notranslate"><span class="pre">--ignore-unclassified</span></code> to ignore the unknown annotation from leaves.</p>
<p>Examples:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># check tree with unknown taxa</span>
<span class="p">(</span><span class="n">Taxa_1</span><span class="p">:</span><span class="mi">1</span><span class="p">,(</span><span class="n">RS_GCF_000019605</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">,(</span><span class="n">Taxa_2</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span><span class="n">GB_GCA_003344655</span><span class="mf">.1</span><span class="p">:</span><span class="mi">1</span><span class="p">):</span><span class="mf">0.5</span><span class="p">):</span><span class="mf">0.5</span><span class="p">);</span>

<span class="c1"># normal way to annotate tree will cause unknown annotation</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">missing_gtdb_v202</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">gtdb</span> \
<span class="o">--</span><span class="n">gtdb</span><span class="o">-</span><span class="n">version</span> <span class="mi">202</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">missing_gtdb_v202_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;Unknown&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;None&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_2|GB_GCA_003344655.1|RS_GCF_000019605.1|Taxa_1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;None&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;Unknown&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span>
<span class="p">}</span>

<span class="c1"># now adding --ignore-unclassified</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">missing_gtdb_v202</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">--</span><span class="n">taxon</span><span class="o">-</span><span class="n">column</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">taxadb</span> <span class="n">gtdb</span> \
<span class="o">--</span><span class="n">gtdb</span><span class="o">-</span><span class="n">version</span> <span class="mi">202</span> \
<span class="o">--</span><span class="n">ignore</span><span class="o">-</span><span class="n">unclassified</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">missing_gtdb_v202_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;evoltype&#39;</span><span class="p">:</span> <span class="s1">&#39;S&#39;</span><span class="p">,</span>
<span class="s1">&#39;lca&#39;</span><span class="p">:</span> <span class="s1">&#39;superkingdom-d__Archaea|phylum-p__Thermoproteota|class-c__Korarchaeia|order-o__Korarchaeales|family-f__Korarchaeaceae|genus-g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;root|d__Archaea|p__Thermoproteota|c__Korarchaeia|o__Korarchaeales|f__Korarchaeaceae|g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;genus&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1|RS_GCF_000019605.1|GB_GCA_003344655.1|Taxa_2&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;g__Korarchaeum&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">1.0</span><span class="p">,</span>
<span class="s1">&#39;common_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;named_lineage&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;rank&#39;</span><span class="p">:</span> <span class="s1">&#39;Unknown&#39;</span><span class="p">,</span>
<span class="s1">&#39;sci_name&#39;</span><span class="p">:</span> <span class="s1">&#39;&#39;</span><span class="p">,</span>
<span class="s1">&#39;species&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span><span class="p">,</span>
<span class="s1">&#39;taxid&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_1&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="functional-annotation">
<h3><a class="toc-backref" href="#id33" role="doc-backlink">Functional annotation</a><a class="headerlink" href="#functional-annotation" title="Link to this heading">¶</a></h3>
<p>Treeprofiler provides module to annotate the tree leaf with functional annotation and sequence alignment file, such as eggnog-mapper output and multiple sequence alignments</p>
<p>In the following session we use example in <code class="docutils literal notranslate"><span class="pre">examples/pratical_example/emapper</span></code></p>
<section id="annotating-multiple-sequence-alignments-msa">
<h4><a class="toc-backref" href="#id34" role="doc-backlink">Annotating Multiple Sequence Alignments(MSA)</a><a class="headerlink" href="#annotating-multiple-sequence-alignments-msa" title="Link to this heading">¶</a></h4>
<p>treeprofiler can anntotate msa to tree and automatically calculate the consesus sequence in the internal node (fixed threshold 0.7), alignment will stored in nodes with property name <cite>alignment</cite>. Using <cite>–column-summary-method alignment=none</cite> can switch off the function for calculating consensus sequence for internal nodes.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># annotate alignment</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> <span class="o">--</span><span class="n">tree</span> <span class="n">nifH</span><span class="o">.</span><span class="n">nw</span> <span class="o">--</span><span class="n">alignment</span> <span class="n">nifH</span><span class="o">.</span><span class="n">faa</span><span class="o">.</span><span class="n">aln</span>

<span class="c1"># mute consensus sequence</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">--</span><span class="n">tree</span> <span class="n">nifH</span><span class="o">.</span><span class="n">nw</span> \
<span class="o">--</span><span class="n">alignment</span> <span class="n">nifH</span><span class="o">.</span><span class="n">faa</span><span class="o">.</span><span class="n">aln</span> \
<span class="o">--</span><span class="n">column</span><span class="o">-</span><span class="n">summary</span><span class="o">-</span><span class="n">method</span> <span class="n">alignment</span><span class="o">=</span><span class="n">none</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>
</pre></div>
</div>
</section>
<section id="annotating-eggnog-mapper-output-and-domain-information">
<h4><a class="toc-backref" href="#id35" role="doc-backlink">Annotating EggNOG-Mapper output and Domain information</a><a class="headerlink" href="#annotating-eggnog-mapper-output-and-domain-information" title="Link to this heading">¶</a></h4>
<p>[EggNOG-mapper](<a class="reference external" href="http://eggnog-mapper.embl.de/">http://eggnog-mapper.embl.de/</a>), is a tool for fast functional annotation of novel sequences. It uses precomputed orthologous groups and phylogenies from the eggNOG database (<a class="reference external" href="http://eggnog5.embl.de">http://eggnog5.embl.de</a>) to transfer functional information from fine-grained orthologs only.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--emapper-annotations</span> <span class="pre">EMAPPER_ANNOTATIONS</span></code></p></td>
<td><p>Attach eggNOG-mapper output out.emapper.annotations</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--emapper-pfam</span> <span class="pre">EMAPPER_PFAM</span></code></p></td>
<td><p>Attach eggNOG-mapper pfam output out.emapper.pfams</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--emapper-smart</span> <span class="pre">EMAPPER_SMART</span></code></p></td>
<td><p>Attach eggNOG-mapper smart output out.emapper.smart</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--alignment</span> <span class="pre">ALIGNMENT</span></code></p></td>
<td><p>Sequence alignment, .fasta format</p></td>
</tr>
</tbody>
</table>
<p>It generates three kind of ouput file,</p>
<ol class="arabic simple">
<li><p>Raw standard output, <code class="docutils literal notranslate"><span class="pre">*.out.emapper.annotations</span></code>, that contains functional annotations and prthology predictions, for example:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">## Mon Feb 27 09:05:50 2023</span>
<span class="c1">## emapper-2.1.9</span>
<span class="c1">## /data/shared/home/emapper/miniconda3/envs/eggnog-mapper-2.1/bin/emapper.py --cpu 20 --mp_start_method forkserver --data_dir /dev/shm/ -o out --output_dir /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j --temp_dir /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j --override -m diamond --dmnd_ignore_warnings --dmnd_algo ctg -i /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j/queries.fasta --evalue 0.001 --score 60 --pident 40 --query_cover 20 --subject_cover 20 --itype proteins --tax_scope auto --target_orthologs all --go_evidence non-electronic --pfam_realign denovo --num_servers 2 --report_orthologs --decorate_gff yes --excel</span>
<span class="c1">##</span>
<span class="c1">#query      seed_ortholog   evalue  score   eggNOG_OGs      max_annot_lvl   COG_category    Description     Preferred_name  GOs     EC      KEGG_ko KEGG_Pathway    KEGG_Module     KEGG_Reaction   KEGG_rclass     BRITE   KEGG_TC CAZy    BiGG_Reaction   PFAMs</span>
<span class="o">....</span>
<span class="c1">## 272 queries scanned</span>
<span class="c1">## Total time (seconds): 45.73449420928955</span>
<span class="c1">## Rate: 5.95 q/s</span>
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>[Pfam](<a class="reference external" href="http://pfam.xfam.org/">http://pfam.xfam.org/</a>) domain annotations, <code class="docutils literal notranslate"><span class="pre">*.out.emapper.pfam</span></code>, for example:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">## Mon Feb 27 09:05:52 2023</span>
<span class="c1">## emapper-2.1.9</span>
<span class="c1">## /data/shared/home/emapper/miniconda3/envs/eggnog-mapper-2.1/bin/emapper.py --cpu 20 --mp_start_method forkserver --data_dir /dev/shm/ -o out --output_dir /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j --temp_dir /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j --override -m diamond --dmnd_ignore_warnings --dmnd_algo ctg -i /emapper_web_jobs/emapper_jobs/user_data/MM_knn6rw6j/queries.fasta --evalue 0.001 --score 60 --pident 40 --query_cover 20 --subject_cover 20 --itype proteins --tax_scope auto --target_orthologs all --go_evidence non-electronic --pfam_realign denovo --num_servers 2 --report_orthologs --decorate_gff yes --excel</span>
<span class="c1">##</span>
<span class="c1"># query_name        hit     evalue  sum_score       query_length    hmmfrom hmmto   seqfrom seqto   query_coverage</span>
<span class="o">...</span>
<span class="c1">## 272 queries scanned</span>
<span class="c1">## Total time (seconds): 28.74908423423767</span>
<span class="c1">## Rate: 9.46 q/s</span>
</pre></div>
</div>
<ol class="arabic simple" start="3">
<li><p>[SMART](<a class="reference external" href="http://smart.embl-heidelberg.de/">http://smart.embl-heidelberg.de/</a>) domain annotation, <code class="docutils literal notranslate"><span class="pre">*.out.emapper.smart.out</span></code>, for example:</p></li>
</ol>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">10020.</span><span class="n">ENSDORP00000023664</span>    <span class="n">MAGE_N</span>  <span class="mi">10</span>      <span class="mi">63</span>      <span class="mf">220000.115599899</span>
<span class="mf">10020.</span><span class="n">ENSDORP00000023664</span>    <span class="n">PTN</span>     <span class="mi">44</span>      <span class="mi">128</span>     <span class="mf">683.160049964146</span>
<span class="mf">10020.</span><span class="n">ENSDORP00000023664</span>    <span class="n">Ephrin_rec_like</span> <span class="mi">73</span>      <span class="mi">117</span>     <span class="mf">248282.169266432</span>
<span class="mf">10020.</span><span class="n">ENSDORP00000023664</span>    <span class="n">PreSET</span>  <span class="mi">87</span>      <span class="mi">186</span>     <span class="mf">494.036044144428</span>
<span class="o">....</span>
</pre></div>
</div>
<dl class="simple">
<dt>TreeProfiler allows users annotate EggNOG-mapper  standard output to target tree with following arguments</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">--emapper-annotations</span></code>, attach eggNOG-mapper output <code class="docutils literal notranslate"><span class="pre">out.emapper.annotations</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--emapper-pfam</span></code>, attach eggNOG-mapper pfam output <code class="docutils literal notranslate"><span class="pre">out.emapper.pfams</span></code>.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">--emapper-smart</span></code>, attach eggNOG-mapper smart output <code class="docutils literal notranslate"><span class="pre">out.emapper.smart</span></code>.</p></li>
</ul>
</dd>
</dl>
<table class="docutils align-default" id="id1">
<caption><span class="caption-text">emapper annotation output and the summary method</span><a class="headerlink" href="#id1" title="Link to this table">¶</a></caption>
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>datatype</p></th>
<th class="head"><p>summary method</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>seed_ortholog</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>evalue</p></td>
<td><p>float</p></td>
<td><p>descriptive stat</p></td>
</tr>
<tr class="row-even"><td><p>score</p></td>
<td><p>float</p></td>
<td><p>descriptive stat</p></td>
</tr>
<tr class="row-odd"><td><p>eggNOG_OGs</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>max_annot_lvl</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>COG_category</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>Description</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>Preferred_name</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>GOs</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>EC</p></td>
<td><p>str</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>KEGG_ko</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>KEGG_Pathway</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>KEGG_Module</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>KEGG_Reaction</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>KEGG_rclass</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>BRITE</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>KEGG_TC</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>CAZy</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-even"><td><p>BiGG_Reaction</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
<tr class="row-odd"><td><p>PFAMs</p></td>
<td><p>list</p></td>
<td><p>counter</p></td>
</tr>
</tbody>
</table>
<p>[check EggNOG-mapper annotation example](#demo2-explore-eggnog-mapper-annotations-data-with-taxonomic-annotation)</p>
</section>
</section>
<section id="analytical-methods-in-treeprofiler">
<h3><a class="toc-backref" href="#id36" role="doc-backlink">Analytical methods in treeProfiler</a><a class="headerlink" href="#analytical-methods-in-treeprofiler" title="Link to this heading">¶</a></h3>
<p>we use examples in <code class="docutils literal notranslate"><span class="pre">examples/analytic_example</span></code></p>
<section id="ancestral-character-reconstruction">
<h4><a class="toc-backref" href="#id37" role="doc-backlink">Ancestral character reconstruction</a><a class="headerlink" href="#ancestral-character-reconstruction" title="Link to this heading">¶</a></h4>
<p>treeprofiler has integrated [pastml](<a class="reference external" href="https://github.com/evolbioinfo/pastml">https://github.com/evolbioinfo/pastml</a>), a flexible platform for ancestral reconstruction with tree with –reconstruct flag in annotate step such as:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--acr-discrete-columns</span> <span class="pre">ACR_DISCRETE_COLUMNS</span> <span class="pre">[ACR_DISCRETE_COLUMNS</span> <span class="pre">...]</span></code></p></td>
<td><p>names of columns to perform acr analysis for discrete traits</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--prediction-method</span> <span class="pre">{MPPA,MAP,JOINT,DOWNPASS,ACCTRAN,DELTRAN,COPY,ALL,ML,MP}</span></code></p></td>
<td><p>Prediction method for ACR discrete analysis. Options: MPPA, MAP, JOINT, DOWNPASS, ACCTRAN, DELTRAN, COPY, ALL, ML, MP. <cite>[Default: MPPA]</cite></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--model</span> <span class="pre">{JC,</span> <span class="pre">F81,</span> <span class="pre">EFT,</span> <span class="pre">HKY,</span> <span class="pre">JTT}</span></code></p></td>
<td><p>Evolutionary model for ML methods in ACR discrete analysis. Options: JC, F81, EFT, HKY, JTT. <cite>[Default: F81]</cite></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--threads</span> <span class="pre">THREADS</span></code></p></td>
<td><p>Number of threads to use for annotation. <cite>[Default: 4]</cite></p></td>
</tr>
</tbody>
</table>
<p>Example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span>
<span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax</span><span class="o">.</span><span class="n">nwk</span> <span class="n">metadata_tab</span><span class="o">.</span><span class="n">csv</span>

<span class="c1"># check metadata</span>
<span class="n">head</span> <span class="n">metadata_tab</span><span class="o">.</span><span class="n">csv</span>
<span class="nb">id</span>  <span class="n">Country</span>
<span class="mi">98</span><span class="n">CMAJ6932</span>  <span class="n">Africa</span>
<span class="mi">98</span><span class="n">CMAJ6933</span>  <span class="n">Africa</span>
<span class="mi">96</span><span class="n">CMAJ6134</span>  <span class="n">Africa</span>
<span class="mi">00</span><span class="n">SEAY5240</span>  <span class="n">WestEurope</span>
<span class="mi">97</span><span class="n">CDAF6240</span>  <span class="n">Africa</span>
<span class="mi">97</span><span class="n">CDAF6238</span>  <span class="n">Africa</span>

<span class="c1"># quick running using all default setting</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax_annotated</span><span class="o">.</span><span class="n">nw</span> \
<span class="o">--</span><span class="n">internal</span><span class="o">-</span><span class="n">parser</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">acr</span><span class="o">-</span><span class="n">discrete</span><span class="o">-</span><span class="n">columns</span> <span class="n">Country</span>  \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="c1"># check properties</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;ROOT&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.0</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa--50||Albania--31||EastEurope--10||Greece--39||WestEurope--22&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="mi">97</span><span class="n">CDAF6238</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;97CDAF6238&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.08034</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span>
<span class="p">}</span>

<span class="c1"># check output files</span>
<span class="n">head</span> <span class="n">marginal_probabilities</span><span class="o">.</span><span class="n">character_Country</span><span class="o">.</span><span class="n">model_F81</span><span class="o">.</span><span class="n">tab</span>
<span class="n">node</span>        <span class="n">Africa</span>  <span class="n">Albania</span> <span class="n">EastEurope</span>      <span class="n">Greece</span>  <span class="n">WestEurope</span>
<span class="n">ROOT</span>        <span class="mf">0.9462054466377042</span>      <span class="mf">0.0019142742715016286</span>   <span class="mf">0.011256165797407233</span>    <span class="mf">0.013434856612985015</span>    <span class="mf">0.027189256680401872</span>
<span class="n">node_1</span>      <span class="mf">0.9497450729621073</span>      <span class="mf">0.00018867741670758483</span>  <span class="mf">0.00048818236055906636</span>  <span class="mf">0.001324183303131325</span>    <span class="mf">0.04825388395749479</span>
<span class="n">node_2</span>      <span class="mf">0.9752818930521312</span>      <span class="mf">0.00048506476303705997</span>  <span class="mf">0.015213913144468159</span>    <span class="mf">0.0034043477773810613</span>   <span class="mf">0.0056147812629824085</span>
<span class="n">node_3</span>      <span class="mf">0.9473989345272481</span>      <span class="mf">0.0002801019197914036</span>   <span class="mf">0.0005949760547048478</span>   <span class="mf">0.001965821926394849</span>    <span class="mf">0.04976016557186095</span>
<span class="n">node_4</span>      <span class="mf">0.9384942099527859</span>      <span class="mf">0.0002164578877048098</span>   <span class="mf">0.00043984526187224396</span>  <span class="mf">0.00151915289715353</span>     <span class="mf">0.05933033400048369</span>
<span class="mi">00</span><span class="n">CZAY4286</span>  <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">1.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>
<span class="n">node_5</span>      <span class="mf">0.9999517018762923</span>      <span class="mf">9.117741186968884e-07</span>   <span class="mf">3.0195194146220156e-05</span>  <span class="mf">6.458698485629717e-06</span>   <span class="mf">1.0732456957024559e-05</span>
<span class="mi">97</span><span class="n">CDAF6238</span>  <span class="mf">1.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>
<span class="mi">94</span><span class="n">CYAF6237</span>  <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">0.0</span>     <span class="mf">1.0</span>

<span class="c1"># check output files</span>
<span class="n">head</span> <span class="n">params</span><span class="o">.</span><span class="n">character_Country</span><span class="o">.</span><span class="n">method_MPPA</span><span class="o">.</span><span class="n">model_F81</span><span class="o">.</span><span class="n">tab</span>
<span class="n">parameter</span>   <span class="n">value</span>
<span class="n">pastml_version</span>      <span class="mf">1.9.42</span>
<span class="n">character</span>   <span class="n">Country</span>
<span class="n">log_likelihood</span>      <span class="o">-</span><span class="mf">118.96060539505257</span>
<span class="n">log_likelihood_restricted_JOINT</span>     <span class="o">-</span><span class="mf">123.17363108674806</span>
<span class="n">log_likelihood_restricted_MAP</span>       <span class="o">-</span><span class="mf">123.3244296265415</span>
<span class="n">log_likelihood_restricted_MPPA</span>      <span class="o">-</span><span class="mf">120.52779174042388</span>
<span class="n">num_scenarios</span>       <span class="mi">96</span>
<span class="n">num_states_per_node_avg</span>     <span class="mf">1.023102310231023</span>
<span class="n">num_unresolved_nodes</span>        <span class="mi">6</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">--acr-discrete-columns</span> <span class="pre">&lt;PROP&gt;</span></code>  allow users to calculate the ancestral character state construction via pastml package. Hence the internal node will be infered the state based on the children leaf node metadata. Users can choose the prediction method using <code class="docutils literal notranslate"><span class="pre">--prediction-method</span> <span class="pre">&lt;METHOD&gt;</span></code>. It will generate the output config file from PASTML package as</p>
<p><code class="docutils literal notranslate"><span class="pre">params.character_{prop}.method_{method}.model_{model}.tab</span></code> which contains information of likelihood from different model/method.</p>
<p><strong>MAXIMUM LIKELIHOOD (ML) METHODS</strong></p>
<p>ML approaches are based on probabilistic models of character evolution along tree branches. From a theoretical standpoint, ML methods have some optimality guaranty [Zhang and Nei, 1997, Gascuel and Steel, 2014], at least in the absence of model violation. Noted that running this ML method will generate output file as <code class="docutils literal notranslate"><span class="pre">marginal_probabilities.character_{prop}.model_{model}.tab</span></code> which contain the calculated propabilities of each character in every internal nodes. Instead <strong>MP method</strong> won’t generate it because it doesn’t compute the marginal propabilities</p>
<p>We provide three ML methods: maximum a posteriori (MAP), Joint, and marginal posterior probabilities approximation (MPPA, recommended):</p>
<ul class="simple">
<li><p><strong>MAP</strong></p></li>
</ul>
<p>(maximum a posteriori). It computes the marginal posterior probabilities of every state for each of the tree nodes, based on the information from the whole tree, i.e. tip states and branch lengths (obtained via two tree traversals: bottom-up, and then top-down). MAP then chooses a state with the highest posterior probability for each node, independently from one node to another. This could induce globally inconsistent scenarios (typically: two very close nodes with incompatible predictions).</p>
<ul class="simple">
<li><p><strong>JOINT</strong></p></li>
</ul>
<p>While MAP chooses predicted states based on all possible scenarios, Joint method [Pupko et al., 2000] reconstructs the states of the scenario with the highest likelihood.</p>
<ul class="simple">
<li><p><strong>MPPA</strong> (default)</p></li>
</ul>
<p>MAP and Joint methods choose one state per node and do not reflect the fact that with real data and large trees, billions of scenarios may have similar posterior probabilities. Based on the marginal posterior probabilities, MPPA (marginal posterior probabilities approximation) chooses for every node a subset of likely states that minimizes the prediction error measured by the Brier score. It therefore sometimes keeps multiple state predictions per node but only when they have similar and high probabilities. Note however that the states not kept by MPPA might still be significant despite being less probable – to check marginal probabilities of each state on a node consult the output marginal probabilities file (can be downloaded via the button below each compressed visualisation).</p>
<ul class="simple">
<li><p><strong>ML</strong></p></li>
</ul>
<p>All the ML methods for ML</p>
<p><strong>Character evolution models (only in ML methods)</strong></p>
<p>We provide some models of character evolution that differ in the way the equilibrium frequencies of states are calculated: <code class="docutils literal notranslate"><span class="pre">JC</span></code>, <code class="docutils literal notranslate"><span class="pre">F81</span></code> <strong>(recommended)</strong>, and <code class="docutils literal notranslate"><span class="pre">EFT</span></code> (estimate-from-tips, <em>not recommended</em>). Using <code class="docutils literal notranslate"><span class="pre">--prediction-method</span> <span class="pre">&lt;model&gt;</span></code> to set up.</p>
<ul class="simple">
<li><p><strong>JC</strong></p></li>
</ul>
<p>With JC model [Jukes and Cantor, 1969] all frequencies, and therefore rates of changes from state i to state j (i ≠ j) are equal.</p>
<ul class="simple">
<li><p><strong>F81</strong> (recommended)</p></li>
</ul>
<p>With F81 model [Felsenstein, 1981], the rate of changes from i to j (i ≠ j) is proportional to the equilibrium frequency of j. The equilibrium frequencies are optimised.</p>
<ul class="simple">
<li><p><strong>EFT</strong></p></li>
</ul>
<p>With EFT (estimate-from-tips) model, the equilibrium frequencies are calculated based on the tip state proportions, the rate of changes from i to j (i ≠ j) is proportional to the equilibrium frequency of j.</p>
<p><strong>MAXIMUM PARSIMONY (MP) METHODS</strong></p>
<p>MP methods aim to minimize the number of state changes in the tree. They are very quick but not very accurate, e.g. they do not take into account branch lengths. We provide three MP methods: <code class="docutils literal notranslate"><span class="pre">DOWNPASS</span></code>, <code class="docutils literal notranslate"><span class="pre">ACCTRAN</span></code>, and <code class="docutils literal notranslate"><span class="pre">DELTRAN</span></code>.</p>
<ul class="simple">
<li><p><strong>DOWNPASS</strong></p></li>
</ul>
<p>DOWNPASS [Maddison and Maddison, 2003] performs two tree traversals: bottom-up and top-down, at the end of which it calculates the most parsimonious states of ancestral nodes based on the information from the whole tree. However some of the nodes might be not completely resolved due to multiple parsimonious solutions.</p>
<ul class="simple">
<li><p><strong>DELTRAN</strong></p></li>
</ul>
<p>DELTRAN (delayed transformation) [Swofford and Maddison, 1987] reduces the number of node state ambiguities by making the changes as close to the tips as possible, hence prioritizing parallel mutations.</p>
<ul class="simple">
<li><p><strong>ACCTRAN</strong></p></li>
</ul>
<p>ACCTRAN (accelerated transformation) [Farris, 1970] reduces the number of node state ambiguities by forcing the state changes to be performed as close to the root as possible, and therefore prioritises the reverse mutations.</p>
<ul class="simple">
<li><p><strong>MP</strong></p></li>
</ul>
<p>all the MP methods for MP</p>
<p>Examples:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># using different model</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax</span><span class="o">.</span><span class="n">nwk</span> \
<span class="o">--</span><span class="n">internal</span><span class="o">-</span><span class="n">parser</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">metadata_tab</span><span class="o">.</span><span class="n">csv</span> \
<span class="o">--</span><span class="n">acr</span><span class="o">-</span><span class="n">discrete</span><span class="o">-</span><span class="n">columns</span> <span class="n">Country</span> \
<span class="o">--</span><span class="n">prediction</span><span class="o">-</span><span class="n">method</span> <span class="n">MPPA</span> \
<span class="o">--</span><span class="n">model</span> <span class="n">JC</span> \
<span class="o">--</span><span class="n">threads</span> <span class="mi">6</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;ROOT&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.0</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa--50||Albania--31||EastEurope--10||Greece--39||WestEurope--22&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;97CDAF6238&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.08034</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span>
<span class="p">}</span>

<span class="c1"># using MP methods (no calculation of ancestral propababilities)</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax</span><span class="o">.</span><span class="n">nwk</span> \
<span class="o">--</span><span class="n">internal</span><span class="o">-</span><span class="n">parser</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">metadata_tab</span><span class="o">.</span><span class="n">csv</span> \
<span class="o">--</span><span class="n">acr</span><span class="o">-</span><span class="n">discrete</span><span class="o">-</span><span class="n">columns</span> <span class="n">Country</span> \
<span class="o">--</span><span class="n">prediction</span><span class="o">-</span><span class="n">method</span> <span class="n">DOWNPASS</span> \
<span class="o">--</span><span class="n">threads</span> <span class="mi">6</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>


<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;ROOT&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.0</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa--50||Albania--31||EastEurope--10||Greece--39||WestEurope--22&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;97CDAF6238&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.08034</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="phylogenetic-signal-delta-statistic">
<h4><a class="toc-backref" href="#id38" role="doc-backlink">Phylogenetic signal delta statistic</a><a class="headerlink" href="#phylogenetic-signal-delta-statistic" title="Link to this heading">¶</a></h4>
<p>Running signal delta statistic required running Ancestral Character Reconstruction using MPPA or MP methods in order to have the ancestral character propabilities. Calculated delta statistic metric and p_value of given trait will be stored in root node as properties.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--delta-stats</span></code></p></td>
<td><p>Calculate delta statistic for discrete traits in ACR analysis, ONLY for MPPA or MAP prediction method. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">False]</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--ent-type</span> <span class="pre">{LSE,SE,GINI}</span></code></p></td>
<td><p>Entropy method to measure the degree of phylogenetic signal between discrete trait and phylogeny. Options: <code class="docutils literal notranslate"><span class="pre">LSE</span></code>, <code class="docutils literal notranslate"><span class="pre">SE</span></code>, <code class="docutils literal notranslate"><span class="pre">GINI</span></code>. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">SE]</span></code> for Shannon Entropy, other options are GINI for Gini impurity and LSE for Linear Shannon Entropy.</p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--iteration</span> <span class="pre">ITERATION</span></code></p></td>
<td><p>Number of iterations for delta statistic calculation. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">10000]</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--lambda0</span> <span class="pre">LAMBDA0</span></code></p></td>
<td><p>Rate parameter of the delta statistic calculation. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">0.1]</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--se</span> <span class="pre">SE</span></code></p></td>
<td><p>Standard deviation of the delta statistic calculation. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">0.5]</span></code></p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--thin</span> <span class="pre">THIN</span></code></p></td>
<td><p>Keep only each xth iterate. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">10]</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--burn</span> <span class="pre">BURN</span></code></p></td>
<td><p>Burned-in iterates. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">100]</span></code></p></td>
</tr>
</tbody>
</table>
<p>Delta statistic Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax</span><span class="o">.</span><span class="n">nwk</span> \
<span class="o">--</span><span class="n">internal</span><span class="o">-</span><span class="n">parser</span> <span class="n">name</span> \
<span class="o">--</span><span class="n">metadata</span> <span class="n">metadata_tab</span><span class="o">.</span><span class="n">csv</span> \
<span class="c1"># acr to obtain propabilities</span>
<span class="o">--</span><span class="n">acr</span><span class="o">-</span><span class="n">discrete</span><span class="o">-</span><span class="n">columns</span> <span class="n">Country</span> \
<span class="o">--</span><span class="n">prediction</span><span class="o">-</span><span class="n">method</span> <span class="n">MPPA</span> \
<span class="o">--</span><span class="n">model</span> <span class="n">F81</span> \
<span class="c1"># delta statistic</span>
<span class="o">--</span><span class="n">delta</span><span class="o">-</span><span class="n">stats</span> \
<span class="o">--</span><span class="n">ent</span><span class="o">-</span><span class="nb">type</span> <span class="n">SE</span> \
<span class="o">--</span><span class="n">iteration</span> <span class="mi">10000</span> \
<span class="o">--</span><span class="n">lambda0</span> <span class="mf">0.1</span> \
<span class="o">--</span><span class="n">se</span> <span class="mf">0.5</span> \
<span class="o">--</span><span class="n">thin</span> <span class="mi">10</span> \
<span class="o">--</span><span class="n">burn</span> <span class="mi">100</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="c1"># delta metric and p_val stored in root node</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">Albanian</span><span class="o">.</span><span class="n">tree</span><span class="mf">.152</span><span class="n">tax_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;ROOT&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.0</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa--50||Albania--31||EastEurope--10||Greece--39||WestEurope--22&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_delta&#39;</span><span class="p">:</span> <span class="s1">&#39;19.52340888828994&#39;</span><span class="p">,</span>
<span class="s1">&#39;Country_pval&#39;</span><span class="p">:</span> <span class="s1">&#39;0.0&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;97CDAF6238&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.08034</span><span class="p">,</span>
<span class="s1">&#39;Country&#39;</span><span class="p">:</span> <span class="s1">&#39;Africa&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="lineage-specificity-analysis">
<h4><a class="toc-backref" href="#id39" role="doc-backlink">Lineage specificity analysis</a><a class="headerlink" href="#lineage-specificity-analysis" title="Link to this heading">¶</a></h4>
<p>Using <code class="docutils literal notranslate"><span class="pre">--ls-columns</span> <span class="pre">&lt;prop_name&gt;</span></code> to start the lineage specificity analysis of boolean traits, the given trait need to be boolean value such as <code class="docutils literal notranslate"><span class="pre">True`;</span> <span class="pre">``False</span></code>; <code class="docutils literal notranslate"><span class="pre">yes</span></code>; <code class="docutils literal notranslate"><span class="pre">no</span></code>; <code class="docutils literal notranslate"><span class="pre">t</span></code>; <code class="docutils literal notranslate"><span class="pre">f</span></code>; <code class="docutils literal notranslate"><span class="pre">1</span></code>; <code class="docutils literal notranslate"><span class="pre">0</span></code>;  which fit the criteria in treeprofiler annotate. Calculated results will be stored in each internal nodes with suffix of <code class="docutils literal notranslate"><span class="pre">_prec</span></code> , <code class="docutils literal notranslate"><span class="pre">_sens</span></code> and <code class="docutils literal notranslate"><span class="pre">_f1</span></code>.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Argument</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--ls-columns</span> <span class="pre">LS_COLUMNS</span> <span class="pre">[LS_COLUMNS</span> <span class="pre">...]</span></code></p></td>
<td><p>names of properties to perform lineage specificity analysis.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="docutils literal notranslate"><span class="pre">--prec-cutoff</span> <span class="pre">PREC_CUTOFF</span></code></p></td>
<td><p>Precision cutoff for lineage specificity analysis. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">0.95]</span></code></p></td>
</tr>
<tr class="row-even"><td><p><code class="docutils literal notranslate"><span class="pre">--sens-cutoff</span> <span class="pre">SENS_CUTOFF</span></code></p></td>
<td><p>Sensitivity threshold for lineage specificity analysis. <code class="docutils literal notranslate"><span class="pre">[Default:</span> <span class="pre">0.95]</span></code></p></td>
</tr>
</tbody>
</table>
<p>Examples:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># in the example we loose the cutoff to 0.5</span>
<span class="n">treeprofiler</span> <span class="n">annotate</span> \
<span class="o">-</span><span class="n">t</span> <span class="n">demo2</span><span class="o">.</span><span class="n">tree</span> \
<span class="o">-</span><span class="n">d</span> <span class="n">demo2_ls</span><span class="o">.</span><span class="n">tsv</span> \
<span class="o">--</span><span class="n">ls</span><span class="o">-</span><span class="n">columns</span> <span class="n">profile1</span> \
<span class="o">--</span><span class="n">prec</span><span class="o">-</span><span class="n">cutoff</span> <span class="mf">0.5</span> \
<span class="o">--</span><span class="n">sens</span><span class="o">-</span><span class="n">cutoff</span> <span class="mf">0.5</span> \
<span class="o">-</span><span class="n">o</span> <span class="o">./</span>

<span class="c1"># check properties</span>
<span class="n">python</span> <span class="n">show_tree_props</span><span class="o">.</span><span class="n">py</span> <span class="n">demo2_annotated</span><span class="o">.</span><span class="n">nw</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">internal</span> <span class="n">node</span> <span class="n">Root</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">properties</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Root&#39;</span><span class="p">,</span>
<span class="s1">&#39;profile1_counter&#39;</span><span class="p">:</span> <span class="s1">&#39;False--33||True--7&#39;</span><span class="p">,</span>
<span class="s1">&#39;profile1_f1&#39;</span><span class="p">:</span> <span class="s1">&#39;0.2978723404255319&#39;</span><span class="p">,</span>
<span class="s1">&#39;profile1_prec&#39;</span><span class="p">:</span> <span class="s1">&#39;0.175&#39;</span><span class="p">,</span>
<span class="s1">&#39;profile1_sens&#39;</span><span class="p">:</span> <span class="s1">&#39;1.0&#39;</span>
<span class="p">}</span>
<span class="n">Target</span> <span class="n">tree</span> <span class="n">leaf</span> <span class="n">node</span> <span class="n">Taxa_0</span> <span class="n">contains</span> <span class="n">the</span> <span class="n">following</span> <span class="n">propertiies</span><span class="p">:</span>
<span class="p">{</span>
<span class="s1">&#39;name&#39;</span><span class="p">:</span> <span class="s1">&#39;Taxa_3&#39;</span><span class="p">,</span>
<span class="s1">&#39;dist&#39;</span><span class="p">:</span> <span class="mf">0.315846</span><span class="p">,</span>
<span class="s1">&#39;profile1&#39;</span><span class="p">:</span> <span class="s1">&#39;False&#39;</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
</section>
<section id="output-formats-for-annotated-trees">
<h3><a class="toc-backref" href="#id40" role="doc-backlink">Output formats for annotated trees</a><a class="headerlink" href="#output-formats-for-annotated-trees" title="Link to this heading">¶</a></h3>
<p>TreeProfiler <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand will generate the following output file</p>
<ol class="arabic simple">
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.nw</em>, newick format with annotated tree</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.ete</em>, ete format with annotated tree</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated_prop2type.txt</em>, config file where store the datatype of each annotated properties</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">&lt;input_tree&gt;</span></code> + <em>_annotated.tsv</em>,  metadata in tab-separated values format with annotated and summarized internal nodes information.</p></li>
</ol>
<p>In the following <code class="docutils literal notranslate"><span class="pre">plot</span></code> step, users can use either <code class="docutils literal notranslate"><span class="pre">.nw</span></code> or <code class="docutils literal notranslate"><span class="pre">.ete</span></code> by putting <code class="docutils literal notranslate"><span class="pre">--input-type</span> <span class="pre">{newick,</span> <span class="pre">ete}</span></code> flag to identify. The difference between <code class="docutils literal notranslate"><span class="pre">.nw</span></code> and <code class="docutils literal notranslate"><span class="pre">.ete</span></code> format is</p>
<blockquote>
<div><ul class="simple">
<li><p>newick file is more universal and be able to used in different other phylogenetic software although associated data of tree nodes will be considered as plain text, so if you use newick format, alongside with the prop2type config file which was generated before by adding <code class="docutils literal notranslate"><span class="pre">--prop2type</span> <span class="pre">&lt;prop2type_file&gt;</span></code></p></li>
<li><p>ete format is a novel format developed to solve the situation we encounter in the previous step, annotated tree can be <strong>recover easily with all the annotated data without changing the data type</strong>. Besides, the ete format optimized the tree file size after mapped with its associated data. Hence it’s very handy for programers in their own script. At this moment we can only view the ete format in treeprofiler, but we will make the ete format more universal to other phylogenetic software. <strong>Hence using ete format in `plot` subcommand is highly reccomended</strong></p></li>
</ul>
</div></blockquote>
</section>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">treeprofiler</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="index.html">Tutorial</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">TreeProfiler Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="#introduction">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="#table-of-contents">Table of Contents</a></li>
<li class="toctree-l2"><a class="reference internal" href="#installation">Installation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li class="toctree-l3"><a class="reference internal" href="#quick-install-via-pip">Quick install via pip</a></li>
<li class="toctree-l3"><a class="reference internal" href="#quick-start-with-examples-dataset">Quick Start with examples dataset</a></li>
<li class="toctree-l3"><a class="reference internal" href="#manual-installation">Manual installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#input-files">Input files</a></li>
<li class="toctree-l3"><a class="reference internal" href="#basic-usage">Basic Usage</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#using-treeprofiler">Using TreeProfiler</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#reading-input-tree">Reading input tree</a></li>
<li class="toctree-l3"><a class="reference internal" href="#treeprofiler-annotate-computing-phylogenetic-profiles-and-annotation"><code class="docutils literal notranslate"><span class="pre">treeprofiler-annotate</span></code> computing phylogenetic profiles and annotation</a></li>
</ul>
</li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  <li><a href="index.html">Tutorial</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Tutorial</a></li>
  </ul></li>
  </ul></li>
</ul>
</div>
<search id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Ziqi Deng & Jaime Huerta-Cepas.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 7.3.7</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="../_sources/documentation/documentations.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>